---
title: "Microcystis colony microbiome analysis with corrigendum"
output: html_notebook
---

The approach was to isolate single Microcystis colonies (100-300 um) into droplets, then sequence the V4 region of all the bacteria in the colony. Colonies were collected in Summer of 2019.

The steps for the OTU building are described in a separate markdown document: LE_H2O2_mothur_OTU_building.Rmd
*Separated out the single colony samples (flagged with the MC2019 sortchem) from the H2O2 mesocosm samples in the shared file.

Loading requirements 
---
Load the required packages:
```{r}
library(ggplot2)
library(vegan)
library(dplyr)
library(tidyr)
library(plotly)
library(indicspecies)
library(patchwork)
library(lubridate)
library(dendextend)
```

Import the Microcystis colony data: 
```{r}
#Load dataframes:
MC_2019.shared <- read.table("MC_2019.shared", header = TRUE, sep="\t")
MC_2019.taxonomy <- read.table("LE_H2O2.taxonomy", header = TRUE, sep="\t")
MC_2019.metadata <- read.table("MC_2019_metadata.txt", header= TRUE, sep="\t")

#Fix up OTU table and merge with taxonomy table:
rownames(MC_2019.shared) <- MC_2019.shared$Group
drop <- c("label", "numOtus", "Group")
MC_2019.shared <- MC_2019.shared[ , !(names(MC_2019.shared) %in% drop)]
MC_2019.shared <- t(MC_2019.shared) #transpose table so that OTUs are row names for merging
MC_2019.shared <- MC_2019.shared[rowSums(MC_2019.shared) != 0,] #Remove Otus that are absent from colony samples
rownames(MC_2019.taxonomy) <- MC_2019.taxonomy$OTU
drop <- "OTU"
MC_2019.taxonomy <- MC_2019.taxonomy[ , !(names(MC_2019.taxonomy) %in% drop)]
MC_2019.taxonomy <- MC_2019.taxonomy[ MC_2019.taxonomy$Size > 2, ] #Only keep OTUs that have more than 2 sequences
MC_2019.merged <- merge(MC_2019.shared, MC_2019.taxonomy, by="row.names", all = FALSE)
#Divide the taxonomy into separate columns for each rank for sorting
MC_2019.merged <- separate(MC_2019.merged, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Convert to percent abundance:
MC_2019.merged[, 2:65] <- apply(MC_2019.merged[,2:65], 2, function(x) (x / sum(x)) *100 )

#Filter dataframe so that we save the control samples into a separate dataframe for later
keep <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Sample_Type == "PBS Blank" ] 
taxcols <- c("Row.names", "Size", "Domain", "Phylum", "Class", "Order", "Family", "Genus")
MC_2019.merged.controls <- MC_2019.merged[ , names(MC_2019.merged) %in% keep | names(MC_2019.merged) %in% taxcols]

#Remove samples that were controls and had a low number of reads:
drop <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Enough_reads == "No" ]
MC_2019.merged <- MC_2019.merged[ , !(names(MC_2019.merged) %in% drop)]

#Remove the Mock Sample:
MC_2019.merged <- MC_2019.merged[ , names(MC_2019.merged) != "MC2019_Mock"]

#How many non-zero values does each OTU have?
otu_nonzero_counts <- apply(MC_2019.merged[2:45], 1, function(y) sum(length(which(y > 0))))
otu_nonzero_counts <- data.frame(value=otu_nonzero_counts)
ggplot(otu_nonzero_counts, aes(x=value)) +
  geom_histogram(binwidth = 1) +
  xlab("Number of non-zero values") +
  ylab("Number of OTUs")
```

The above warning about the dataframe is generated because the Genus name is prepended with a ";" separator, but not followed by any text. However, the data parses correctly, so nothing to worry about.

There are a lot of OTUs that are present in only 1 samples. What is the highest relative abundance of these OTUs?
```{r}
#Add the nonzero sample count of each OTU to the dataframe
MC_2019.merged$Nonzeros <- otu_nonzero_counts$value
#Remove the empty OTUs:
MC_2019.merged <- MC_2019.merged[MC_2019.merged$Nonzeros != 0,]
#Filter out the taxa present in 1 samples, then get the max relative abundance for those taxa
MC_2019.merged.raretaxa <- MC_2019.merged[MC_2019.merged$Nonzeros == 1,]
rare_otu_max_counts <- apply(MC_2019.merged.raretaxa[2:45], 1, max)

#Histogram
full_max_hist <- ggplot(data.frame(value=rare_otu_max_counts), aes(x=value)) +
  geom_histogram(binwidth = 0.2) +
  scale_x_continuous(breaks = seq(0,8, by = 0.05)) +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2)) +
  #coord_cartesian(ylim=(0:300)) +
  xlab("Maximum percent abundance") +
  ylab("Number of rare OTUs")

zoomed_max_hist <- ggplot(data.frame(value=rare_otu_max_counts), aes(x=value)) +
  geom_histogram(binwidth = 0.2) +
  scale_x_continuous(breaks = seq(0,8, by = 0.05)) +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2)) +
  coord_cartesian(ylim=(0:300), xlim=(0:1.5)) +
  xlab("Maximum percent abundance 0-1.5%") +
  ylab("Number of rare OTUs")

full_max_hist
```

Most of the rare OTUs only reach a maximum abundance of 0.25% or less of the reads, with the majority 0.05% or below. Let's remove them from the analysis, but keep any rare taxa that are present at a relative abundance 0.1% or higher.
```{r}
#Get a list of OTUs to remove. They are only present in 1 sample at a maximum relative abundance less than 0.1%
MC_2019.merged.raretaxa$Max_abund <- rare_otu_max_counts
rownames(MC_2019.merged.raretaxa) <- MC_2019.merged.raretaxa$Row.names
rares_to_remove <- rownames(MC_2019.merged.raretaxa[ MC_2019.merged.raretaxa$Max_abund <= 0.1, ])

#Reprocess the data frame, removing these taxa:
#Load dataframes:
MC_2019.shared <- read.table("MC_2019.shared", header = TRUE, sep="\t")
MC_2019.taxonomy <- read.table("LE_H2O2.taxonomy", header = TRUE, sep="\t")
MC_2019.metadata <- read.table("MC_2019_metadata.txt", header= TRUE, sep="\t")

#Fix up OTU table and merge with taxonomy table:
rownames(MC_2019.shared) <- MC_2019.shared$Group
drop <- c("label", "numOtus", "Group")
MC_2019.shared <- MC_2019.shared[ , !(names(MC_2019.shared) %in% drop)]
MC_2019.shared <- t(MC_2019.shared) #transpose table so that OTUs are row names for merging
MC_2019.shared <- MC_2019.shared[rowSums(MC_2019.shared) != 0,] #Remove Otus that are absent from colony samples
rownames(MC_2019.taxonomy) <- MC_2019.taxonomy$OTU
drop <- "OTU"
MC_2019.taxonomy <- MC_2019.taxonomy[ , !(names(MC_2019.taxonomy) %in% drop)]
MC_2019.taxonomy <- MC_2019.taxonomy[ MC_2019.taxonomy$Size > 2, ] #Only keep OTUs that have more than 2 sequences
MC_2019.merged <- merge(MC_2019.shared, MC_2019.taxonomy, by="row.names", all = FALSE)
#Divide the taxonomy into separate columns for each rank for sorting
MC_2019.merged <- separate(MC_2019.merged, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Filter dataframe so that we save the control samples into a separate dataframe for later
keep <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Sample_Type == "PBS Blank" ] 
taxcols <- c("Row.names", "Size", "Domain", "Phylum", "Class", "Order", "Family", "Genus")
MC_2019.merged.controls <- MC_2019.merged[ , names(MC_2019.merged) %in% keep | names(MC_2019.merged) %in% taxcols ]

#Remove samples that were controls and had a low number of reads:
drop <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Enough_reads == "No" ]
MC_2019.merged <- MC_2019.merged[ , !(names(MC_2019.merged) %in% drop)]

#Remove the Mock Sample:
MC_2019.merged <- MC_2019.merged[ , names(MC_2019.merged) != "MC2019_Mock"]

#Calculate the observation frequency of each OTU, add to dataframe:
otu_nonzero_counts <- apply(MC_2019.merged[2:45], 1, function(y) sum(length(which(y > 0))))
otu_obs_freq <- ( otu_nonzero_counts / 44 ) * 100
MC_2019.merged$otu_obs_freq <- otu_obs_freq

#Remove any OTUs not present in the colony samples:
MC_2019.merged <- MC_2019.merged[ MC_2019.merged$otu_obs_freq !=0, ]
Num_otus_prefilter <- nrow(MC_2019.merged)

#Remove the OTUs flagged as rare:
MC_2019.merged <- MC_2019.merged[ !(MC_2019.merged$Row.names %in% rares_to_remove), ]
Num_otus_postfilter <- nrow(MC_2019.merged)

#How many OTUs were removed?
Num_otus_prefilter - Num_otus_postfilter

#What is the percentage of OTUs kept?
(Num_otus_postfilter / Num_otus_prefilter) * 100

#Convert to percent abundance:
MC_2019.merged[, 2:45] <- apply(MC_2019.merged[,2:45], 2, function(x) (x / sum(x)) *100 )

#Convert dataframe to long format:
MC_2019.merged.long <- gather(MC_2019.merged, Sample_ID, Perc_abund, 2:45)

#Add colony metadata:
MC_2019.merged.long <- merge(MC_2019.merged.long, MC_2019.metadata, by="Sample_ID", all.x = TRUE)
drop <- c("Sample_Type", "Amplification", "Enough_reads") #remove these columns, because they are no longer useful
MC_2019.merged.long <- MC_2019.merged.long[, !(names(MC_2019.merged.long) %in% drop) ]
```
We removed 226 OTUs flagged as rare, keeping ~52%.

Top Otus
----
Plot the top 20 Otus, based on mean percent abundance across all colonies:
```{r}
#Plot the distribution of percent abundance of each of the top 20 OTUs, based on mean abundance across all colonies:
MC_2019.merged.long$Row.names = with(MC_2019.merged.long, reorder(Row.names, Perc_abund, mean))
All_Otu_sorted <- unique(sort(MC_2019.merged.long$Row.names, decreasing=TRUE))[1:20]

MC2019_rank_abund_plot <- ggplot(MC_2019.merged.long, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
  geom_jitter(size=0.3, alpha=0.7) +
  stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2),
    legend.position = "none") +
  scale_x_discrete(limits = All_Otu_sorted, labels = c("Microcystis", "Cyanobium Otu4", "Uncultured Sutterellaceae\nOtu43", "Lysinibacillus", "Pseudanabaena Otu7", "Uncultured Microscillaceae\nOtu32", "Sphingomonas Otu343", "Unclassified Bacillaceae\nOtu375", "Unclassified Caulobacteraceae\nOtu34", "Phreatobacter Otu114", "Tabrizicola Otu46", "Uncultured Microscillaceae\nOtu35", "Pseudanabaena Otu11", "Bradyrhizobium Otu602", "Uncultured Microscillaceae\nOtu56", "Nitrosomonadaceae DSD61\nOtu80", "Reyranella Otu312", "Roseomonas Otu342", "Uncultured Microscillaceae\nOtu52", "Roseomonas Otu10")) +
  coord_cartesian(ylim = c(0,100)) +
  ylab("Percent abundance")

MC2019_rank_abund_plot
```

Remake the above plot, but exclude the Microcystis OTU:
```{r}
#Plot the distribution of percent abundance of each of the top 20 OTUs, based on mean abundance across all colonies:
All_Otu_sorted <- unique(sort(MC_2019.merged.long$Row.names, decreasing=TRUE))[2:21]

MC2019_rank_abund_plot_noMC <- ggplot(MC_2019.merged.long, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund, fill=Genus)) +
  geom_jitter(size=0.3, alpha=0.7) +
  stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2),
    legend.position = "none") +
  scale_x_discrete(limits = All_Otu_sorted, labels = c("Cyanobium Otu4", "Uncultured Sutterellaceae\nOtu43", "Lysinibacillus", "Pseudanabaena Otu7", "Uncultured Microscillaceae\nOtu32", "Sphingomonas Otu343", "Unclassified Bacillaceae\nOtu375", "Unclassified Caulobacteraceae\nOtu34", "Phreatobacter Otu114", "Tabrizicola Otu46", "Uncultured Microscillaceae\nOtu35", "Pseudanabaena Otu11", "Bradyrhizobium Otu602", "Uncultured Microscillaceae\nOtu56", "Nitrosomonadaceae DSD61\nOtu80", "Reyranella Otu312", "Roseomonas Otu342", "Uncultured Microscillaceae\nOtu52", "Roseomonas Otu10")) +
  coord_cartesian(ylim = c(0,80)) +
  ylab("Percent abundance")

MC2019_rank_abund_plot_noMC
```

In the above plots, the percent abundance of the 20 most abundant Otus (based on mean percent abundance across all colonies) is shown. Each black dot represents the percent abundance of the listed OTUs in a single colony. The red dot shows the mean percent abundance across all colonies, and the red error bars show the 95% confidence intervals for the mean across all colonies. 

There are indeed heterotrophic bacteria that are present at high relative abundances. However, there are no Otus that are always present (there are a lot of zero counts for each Otu other than Microcystis). How similar are the communities in the colonies across time and colony morphology?

Make the top 20 rank abundance plot, but group the samples by colony morphology:
```{r}
#Plot the distribution of percent abundance of each of the top 20 OTUs, based on mean abundance across all colonies:
MC2019_rank_abund_plot_morph <- ggplot(MC_2019.merged.long, aes(x=Row.names, y=Perc_abund, fill=Morphotype)) +
  geom_boxplot(outlier.alpha = 0) +
  scale_fill_viridis_d(alpha=0.6) +
  geom_jitter(color="black", size=0.05, alpha=0.9) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2)) +
 scale_x_discrete(limits = All_Otu_sorted, 
    labels = c("Cyanobium Otu4", "Uncultured Sutterellaceae\nOtu43", "Lysinibacillus", "Pseudanabaena Otu7", "Uncultured Microscillaceae\nOtu32", "Sphingomonas Otu343", "Unclassified Bacillaceae\nOtu375", "Unclassified Caulobacteraceae\nOtu34", "Phreatobacter Otu114", "Tabrizicola Otu46", "Uncultured Microscillaceae\nOtu35", "Pseudanabaena Otu11", "Bradyrhizobium Otu602", "Uncultured Microscillaceae\nOtu56", "Nitrosomonadaceae DSD61\nOtu80", "Reyranella Otu312", "Roseomonas Otu342", "Uncultured Microscillaceae\nOtu52", "Roseomonas Otu10")) +
  ylab("Percent abundance")

MC2019_rank_abund_plot_morph
```

No obvious patterns stand out here, but there may be too much data to visualize on this kind of plot.

Cluster and ordination analysis
----
Construct an NMDS plot to cluster samples, overlay colony morphology and collection date:
```{r}
#To make an NMDS plot, we need a matrix of OTU abundance where the columns are OTUs and the rows are samples. We can only have abundance information, no metadata.

#Take our processed OTU table and remove the metadata columns:
#Columns 1-47 contain the info we need, the rest is metadata
MC_2019.matrix <- MC_2019.merged[,1:45]
rownames(MC_2019.matrix) <- MC_2019.matrix$Row.names
MC_2019.matrix <- MC_2019.matrix[ , colnames(MC_2019.matrix) != "Row.names" ]
#Check that there are only headers for samples:
names(MC_2019.matrix)
```
```{r}
#Transpose the matrix so that samples are rows and OTUs are columns:
MC_2019.matrix <- t(MC_2019.matrix)

#Remove columns (Otus) that are empty:
MC_2019.matrix <- MC_2019.matrix[  , colSums(MC_2019.matrix) != 0 ]

#Build the plot using the metaNMDS function in vegan with two axes:
set.seed(1)
NMDS <- metaMDS(MC_2019.matrix, distance = "bray", k=2, autotransform = FALSE, try = 40, trymax = 100)
```

**Notes on the stats displayed in NDMS:**  
The distance metric used is Bray-Curtis dissimilarity. It is defined as:  
$$BC_{ij} = 1 - \frac{2C_{ij}}{S_i + S_j}$$  
Where:  
1. *i* and *j* represent samples to be compared  
2. Cij is the sum of the counts for each OTU found in both samples, using only the lesser count from either for each species  
    + For an example, consider two imaginary forests. Forest 1 has 10 squirrels, 3 bears, and a leopard. Forest 2 has 13 squirrels, 1 bear, and no leopards. When counting squirrels for Cij the value would be 10 and for bears it would be 1. Leopards would not contribute to Cij because they are not present in both forests. The value for Cij would be 10 + 1 = 14.  
3. Si and Sj are the sum of all species in samples *i* and *j*, respectively.  
    + Continuing the example from above the Si would be 10 + 3 + 1 = 14 and Sj would be 13 + 1 + 0 = 14.  
    + The Bray-Curtis dissimilarity for the example would be  
      1 - [(2 * 14) / (14 + 14)] = 1 - [28 / 28] = 1 - 1 = 0  

The stress of the NMDS plot indicates the goodness of fit. A stress value 0.1 or lower is considered a fair fit,  
and a stress value below 0.05 is considered a good fit. Anything above 0.2 is considered very poor and above 0.3 is suspect for arbitrary clustering. The stress of the NMDS is sensitive to the number of axes, (set as k in the vegan function).  
Let's see what the effect of adding a third axis is:  
```{r}
#Build the plot using the metaNMDS function in vegan with three axes:
set.seed(2)
NMDS_3 <- metaMDS(MC_2019.matrix, distance = "bray", k=3, autotransform = FALSE, try = 40, trymax = 100)
```

The stress decreased from ~0.11 to ~0.075. Let's contiune to add another axis to see how the stress changes:
```{r}
#Build the plot using the metaNMDS function in vegan with four axes:
set.seed(4)
NMDS_4 <- metaMDS(MC_2019.matrix, distance = "bray", k=4, autotransform = FALSE, try = 40, trymax = 100)
```
Moving from 3 to 4 axes changes the stress from ~0.075 to ~0.052. Interpreting 4 axes will be difficult, and the improvement in stress is marginal so let's start with using 3 axes.

Another important check is to make sure that the ordination distance actually represents the calculated Bray Curtis dissimilarity. We can do this by plotting the ordination distance versus the dissimilarity:  
```{r}
stressplot(NMDS_3, xlim=c(0,1))
```
As expected, as the distance between two points increases on the ordination, the observed dissimilarity increases. 


Our fit looks good, so lets make plots from the NMDS with 3 axes:  
```{r fig.height = 12, fig.width= 10}
#Extract the axis coordinates (the NMDS scores):
NMDS_3.scores <- as.data.frame(scores(NMDS_3))

#Add metadata:
NMDS_3.scores <- merge(NMDS_3.scores, MC_2019.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE)
rownames(NMDS_3.scores) <- NMDS_3.scores$Row.names
drop <- "Row.names"
NMDS_3.scores <- NMDS_3.scores[ , !(names(NMDS_3.scores) %in% drop) ]

#Plot axis 1 vs axis 2:
NMDS_3.1v2.plot <- ggplot(NMDS_3.scores, aes(x=NMDS1, y=NMDS2)) +
  geom_point(size = 8, aes(shape=Morphotype, colour=Date_Collected)) +
  theme(
    axis.title.x = element_text(size=20, margin = margin(t = 20, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=20, margin = margin(t = 0, r = 20, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 20, color = "black", margin = margin(t = 19, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 20, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
    axis.ticks.length = unit(-0.5, "cm"),
    axis.ticks = element_line(size = 1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 1")

#Plot axis 1 vs axis 3:
NMDS_3.1v3.plot <- ggplot(NMDS_3.scores, aes(x=NMDS1, y=NMDS3)) +
  geom_point(size = 8, aes(shape=Morphotype, colour=Date_Collected)) +
  ggtitle("Stress = 0.075") +
  theme(plot.title = element_text(hjust=1, size=20),
    axis.title.x = element_text(size=20, margin = margin(t = 20, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=20, margin = margin(t = 0, r = 20, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 20, color = "black", margin = margin(t = 19, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 20, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
    axis.ticks.length = unit(-0.5, "cm"),
    axis.ticks = element_line(size = 1),
    legend.position = "none") +
  ylab("NMDS 3") +
  xlab("NMDS 1")

#Plot axis 2 vs axis 3:
NMDS_3.2v3.plot <- ggplot(NMDS_3.scores, aes(x=NMDS2, y=NMDS3)) +
  geom_point(size = 8, aes(shape=Morphotype, colour=Date_Collected)) +
  theme(
    axis.title.x = element_text(size=20, margin = margin(t = 20, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=20, margin = margin(t = 0, r = 20, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 20, color = "black", margin = margin(t = 19, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 20, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
    axis.ticks.length = unit(-0.5, "cm"),
    axis.ticks = element_line(size = 1),
    legend.position = "bottom",
    legend.box = "vertical",
    legend.text = element_text(size = 20), legend.title = element_text(size = 25)) +
  ylab("NMDS 3") +
  xlab("NMDS 2")

(NMDS_3.1v2.plot + NMDS_3.1v3.plot) / NMDS_3.2v3.plot
```
There appears to be separation along axis 1 based on morphotype (wesenbergii vs all others), and by sampling date Sep-16 and Sep-9 vs others.  

Let's see if we can make a 3D plot to better visualize all the axes simultaneously:  
```{r}
NMDS_3_3D <- plot_ly(NMDS_3.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = ~Date_Collected, colors = c("coral", "gold", "blue", "aquamarine", "deeppink2", "green3"), symbol = ~Morphotype, symbols = c("circle", "diamond", "square"), showlegend = F) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
          legend = list(title = "Morphotype"),
         annotations = list(text = "Stress = 0.075",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))

NMDS_3_3D
```

The plot is interactive. You can click and drag to view the data from different views of the 3 axes.  
**Legend:**  
**Date**  
blue = 22-Jul-19  
magenta = 5-Aug-19  
gold = 19-Aug-19  
aqua = 3-Sep-19  
green = 9-Sep-19  
coral = 16-Sep-19  

**Morphotype**  
circle = aeruginosa  
diamond = novacekii  
square = wesenbergii  

There appears to be separation based on whether the morphotype is wesenbergii vs novacekki and based on date. However, there is also  
a lot of variation between the colonies in the September 16 colonies that isn't explained by morphotype. The aeruginosa samples also seem to cluster with the novacekii samples.  

Is the separation of samples in th NMDS based on morphotype and date statistically significant?  
We can test this with an analysis of similarity (ANOSIM). It determines if the ranked dissimilarities between groups is significantly different.  
```{r}
#ANOSIM for collection date:
ano_date <- anosim(MC_2019.matrix, NMDS_3.scores$Date_Collected, distance = "bray", permutations = 999)
ano_date
```

The significance score is below 0.05, indicating that the mean ranked dissimilarities between collection dates are significantly different.  
The R statistic compares the mean ranked dissimilarities between groups to the mean of the ranked dissimilarities within groups.  
As R approaches 1.0, the groups are more dissimilar. An R stat of 0 indicates that there is an even distribution of high and low ranks between and within groups. R values below 0 indicate that there is higher dissimilarity within the groups than between them.  
We can conclude that differences in the mean dissimilarity by date are significant, but the effect is relatively weak.  
```{r}
#ANOSIM for colony morphology:
ano_morph <- anosim(MC_2019.matrix, NMDS_3.scores$Morphotype, distance = "bray", permutations = 999)
ano_morph
```

Identifying indicator species for significantly different groups
----
Are there any indicator species that drive the differences based on sample date?  
The indicator species are calculated with the indicspecies R package developed by Caceres and Legendre (2009): https://esajournals.onlinelibrary.wiley.com/doi/full/10.1890/08-1823.1  
```{r}
indval <- multipatt(MC_2019.matrix, NMDS_3.scores$Date_Collected, func = "IndVal.g", control = how(nperm=999))
summary(indval, indvalcomp=TRUE)
```

The function calculates an Indicator Value (IndVal) index between each species and each group (in our case sample date) then looks for  
group corresponding to the highest association value. The significance is then tested with a permutation test. The IndVal is corrected for unequal sampling across groups.  

An explanation of the output:  
The A stat is the specificity or the positive predictive value of the species as an indicator of the group, or the probability that a given sample  
belongs to the target group given the fact that the species has been found.  

The B stat is the fidelity or sensitivity of the species as an indicator of the group, or the probability of finding the species in samples of the group.  

"stat" and "p.value" are the results of the significance test.  

For example, Otu00375 is associated with the 16-Sep-19, and the association is highly significant. A = 1.0 and B = 1.0, meaning that it only occurs in colonies collected on 16-Sep-19 and it occurs on every colony collected on 16-Sep-19. This is a strong indicator species.  

In contrast, Otu01608 has A = 1.0 and B = 0.75. This means that it is only found on colonies collected on 16-Sep-19, but not all of the colonies collected on this date (only ~75% of them).  

Another example: Otu00209 is significantly associated with the 16-Sep-19 group. Its B value is 1.0, which means that it is found in all of the 16-Sep-19 samples. However, A is 0.99 which means it occured in a small number of colonies that were not collected on 16-Sep-19.  

These stats were calculated for every collection date, and each combination of collection dates. Only 16-Sep and 9-Sep samples have really strong indicator species.

Let's plot the mean abundance of the indicator species in their respective groups:   
```{r}
#Get a list of Otus specific to each group:
indval.Otus.16Sep <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,1] == 1 & indval$sign[,9] <= 0.05, ])
indval.Otus.19Aug <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,2] == 1 & indval$sign[,9] <= 0.05, ])
indval.Otus.22Jul <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,3] == 1 & indval$sign[,9] <= 0.05, ])
indval.Otus.3Sep <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,4] == 1 & indval$sign[,9] <= 0.05, ])
indval.Otus.5Aug <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,5] == 1 & indval$sign[,9] <= 0.05, ])
indval.Otus.9Sep <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,6] == 1 & indval$sign[,9] <= 0.05, ])

#Plot for 16-Sep-19:
indval_Otus_16Sep <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.16Sep & Date_Collected == "16-Sep-19")

indval_Otus_16Sep_plot <- ggplot(indval_Otus_16Sep, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 16-Sep-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.5, "cm"),
      axis.ticks = element_line(size = 1),
      legend.position = "none") +
  coord_cartesian(ylim = c(0,60)) +  
  ylab("Percent abundance")

#Plot for 9-Sep-19:
indval_Otus_9Sep <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.9Sep & Date_Collected == "9-Sep-19")
                             
indval_Otus_9Sep_plot <-ggplot(indval_Otus_9Sep, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 9-Sep-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.5, "cm"),
      axis.ticks = element_line(size = 1),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,80)) +
    ylab("Percent abundance")

#Plot for 19-Aug-19:
indval_Otus_19Aug <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.19Aug & Date_Collected == "19-Aug-19")

indval_Otus_19Aug_plot <-ggplot(indval_Otus_19Aug, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 19-Aug-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.5, "cm"),
      axis.ticks = element_line(size = 1),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,1)) +
    ylab("Percent abundance")

#Plot for 3-Sep-19:
indval_Otus_3Sep <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.3Sep & Date_Collected == "3-Sep-19")

indval_Otus_3Sep_plot <- ggplot(indval_Otus_3Sep, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 3-Sep-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.5, "cm"),
      axis.ticks = element_line(size = 1),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,5)) +
    ylab("Percent abundance")

#Plot for 5-Aug-19:
indval_Otus_5Aug <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.5Aug & Date_Collected == "5-Aug-19")

indval_Otus_5Aug_plot <-ggplot(indval_Otus_5Aug, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 5-Aug-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.5, "cm"),
      axis.ticks = element_line(size = 1),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,2)) +
    ylab("Percent abundance")

#Plot for 22-Jul-19:
indval_Otus_22Jul <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.22Jul & Date_Collected == "22-Jul-19")

indval_Otus_22Jul_plot <-ggplot(indval_Otus_22Jul, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 22-Jul-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.5, "cm"),
      axis.ticks = element_line(size = 1),
      legend.position = "none") +
    ylab("Percent abundance")

(indval_Otus_5Aug_plot | indval_Otus_19Aug_plot | indval_Otus_22Jul_plot) / (indval_Otus_3Sep_plot |  indval_Otus_9Sep_plot | indval_Otus_16Sep_plot)
```
It looks like the separation in the NMDS is largely driven by the high abundance of Lysinibacillus and bacillaceae OTUs in the 16-Sep-19 samples, and the 5 colonies with a high relative abundance of Cyanobium Otu 4.  

What are the top 20 Otus for the 9-Sep-19 samples?  
```{r}
All_Otu_Sep_9 <- filter(MC_2019.merged.long, Date_Collected == "9-Sep-19")
All_Otu_Sep_9$Row.names = with(All_Otu_Sep_9, reorder(Row.names, Perc_abund, mean))
All_Otu_Sep_9_sorted <- unique(sort(All_Otu_Sep_9$Row.names, decreasing=TRUE))[1:20]
All_Otu_Sep_9_sorted_tail <- unique(sort(All_Otu_Sep_9$Row.names, decreasing=TRUE))[5:20]

#Plot all 20:
ggplot(All_Otu_Sep_9, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=0.3, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
    theme(
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    scale_x_discrete(limits=All_Otu_Sep_9_sorted) +
    coord_cartesian(ylim = c(0,80)) +
    ylab("Percent abundance")

#Remove the 4 most abundant OTUs to see the tail
ggplot(All_Otu_Sep_9, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=0.3, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
    theme(
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    scale_x_discrete(limits=All_Otu_Sep_9_sorted) +
  coord_cartesian(ylim = c(0,1.5)) +
    ylab("Percent abundance")
```

What are the top 20 Otus for the 16-Sep-19 samples?  
```{r}
All_Otu_Sep_16 <- filter(MC_2019.merged.long, Date_Collected == "16-Sep-19")
All_Otu_Sep_16$Row.names = with(All_Otu_Sep_16, reorder(Row.names, Perc_abund, mean))
All_Otu_Sep_16_sorted <- unique(sort(All_Otu_Sep_16$Row.names, decreasing=TRUE))[1:20]
All_Otu_Sep_16_sorted_tail <- unique(sort(All_Otu_Sep_16$Row.names, decreasing=TRUE))[5:20]

#Plot the top 20 Otus
ggplot(All_Otu_Sep_16, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=0.3, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
    theme(
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    scale_x_discrete(limits=All_Otu_Sep_16_sorted) +
    coord_cartesian(ylim = c(0,100)) +
    ylab("Percent abundance")

#Plot without the 4 most abundant Otus:
ggplot(All_Otu_Sep_16, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=0.3, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
    theme(
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    scale_x_discrete(limits=All_Otu_Sep_16_sorted_tail) +
    coord_cartesian(ylim = c(0,25)) +
    ylab("Percent abundance")
```


Screening for contaminants  
----
Let's look at the data from the size fractionated H2O2 mesocosm samples to see if these taxa are indeed associated with communities that pass through the 100 um filter or not.  
```{r}
#Import the dataframes:
LE_H2O2.shared <- read.table("LE_H2O2.shared", header = TRUE, sep="\t")
LE_H2O2.metadata <- read.table("LE_H2O2.metadata.txt", header= TRUE, sep="\t")

#Fix up OTU table and merge with taxonomy table:
rownames(LE_H2O2.shared) <- LE_H2O2.shared$Group
drop <- c("label", "numOtus", "Group")
LE_H2O2.shared <- LE_H2O2.shared[ , !(names(LE_H2O2.shared) %in% drop)]
LE_H2O2.shared <- t(LE_H2O2.shared) #transpose table so that OTUs are row names for merging
LE_H2O2.merged <- merge(LE_H2O2.shared, MC_2019.taxonomy, by="row.names", all = FALSE)
#Divide the taxonomy into separate columns for each rank for sorting
LE_H2O2.merged <- separate(LE_H2O2.merged, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Remove samples that were controls and had a low number of reads:
drop <- LE_H2O2.metadata$Sortchem[ LE_H2O2.metadata$Enough_reads == "No" ]
LE_H2O2.merged <- LE_H2O2.merged[ , !(names(LE_H2O2.merged) %in% drop)]

#Remove samples that were cross-contaminted during plate loading:
drop <- c("E2019_1152_1", "E2019_1177_1")
LE_H2O2.merged <- LE_H2O2.merged[ , !(names(LE_H2O2.merged) %in% drop)]

#Sum the T. thermophilus reads in each sample, then remove those OTUs from the data:
LE_H2O2.merged.std_reads <- as.data.frame(colSums(LE_H2O2.merged[2:214][LE_H2O2.merged$Genus == "Thermus",]))
colnames(LE_H2O2.merged.std_reads) <- c("Total_Thermus_reads")
LE_H2O2.merged.std_reads$Sortchem <- row.names(LE_H2O2.merged.std_reads)
LE_H2O2.merged <- LE_H2O2.merged[LE_H2O2.merged$Genus != "Thermus", ]

#Convert dataframe to long format:
LE_H2O2.merged.long <- gather(LE_H2O2.merged, Sortchem, Counts, 2:214)

#Add the tallied Thermus reads to the dataframe:
LE_H2O2.merged.long <- merge(LE_H2O2.merged.long, LE_H2O2.merged.std_reads, by="Sortchem", all = TRUE)

#Add metadata:
LE_H2O2.merged.long <- merge(LE_H2O2.merged.long, LE_H2O2.metadata, by="Sortchem", all = FALSE)

#Calculate absolute abundance:
LE_H2O2.merged.long$Reads_mL <- (LE_H2O2.merged.long$Counts * LE_H2O2.merged.long$Spiked_copy_number) / (LE_H2O2.merged.long$Total_Thermus_reads * LE_H2O2.merged.long$mLs_filtered)
```

Let's see how abundant Cyanobium Otu 4 is in whole water vs free-living samples:  
```{r}
#Get samples from size fractionation experiments:
LE_H2O2.merged.size_frac <- filter(LE_H2O2.merged.long, Experiment_Type == "Size_Frac")

#Calculate mean, sd, se, and 95% CI:
LE_H2O2.merged.size_frac_sum <- LE_H2O2.merged.size_frac %>%
  group_by(Row.names, Exp_Date, Other_Treatment) %>%
  summarise(n=n(), mean=mean(Reads_mL), sd=sd(Reads_mL)) %>%
  mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

#Plot Cyanobium abundance:
Cyanobium_high_plot <- filter(LE_H2O2.merged.size_frac_sum, Row.names == "Otu00004") %>%
  ggplot(aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    scale_x_discrete(limits=c("3-Aug-18", "10-Aug-18", "21-Aug-18", "14-Sep-18")) +
    ylab("Cyanobium Otu4 Abundance (reads/mL)")

Cyanobium_low_plot <- filter(LE_H2O2.merged.size_frac_sum, Row.names == "Otu00004") %>%
  ggplot(aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.title = element_blank()) +
    scale_x_discrete(limits=c("26-Jul-19", "6-Aug-19", "24-Aug-19", "20-Sep-19")) +
    coord_cartesian(ylim=c(0,6000)) +
    ylab("Cyanobium Otu4 Abundance (reads/mL)")

Cyanobium_plot <- Cyanobium_high_plot + Cyanobium_low_plot
ggsave("Cyanobium_plot.pdf", plot = Cyanobium_plot, width = 18, height = 12, units = "cm", dpi=320)
Cyanobium_plot
```

It looks like there is more Cyanobium in the whole water samples than in the 100 um filtered samples on 24-Aug-19 and 20-Sep-19. Is the mean abundance of Cyanobium significantly higher in the whole water vs 100 um filtered water on these dates?    
```{r}
#T-test assuming unequal variance for 24-Aug-19
t.stat.vector.x <- filter(LE_H2O2.merged.size_frac, Row.names == "Otu00004" & Exp_Date == "24-Aug-19" & Other_Treatment == "Whole water")[,23]

t.stat.vector.y <- filter(LE_H2O2.merged.size_frac, Row.names == "Otu00004" & Exp_Date == "24-Aug-19" & Other_Treatment == "100 um filtered")[,23]

t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "greater")
```
```{r}
#T-test assuming unequal variance for 20-Sep-19
t.stat.vector.x <- filter(LE_H2O2.merged.size_frac, Row.names == "Otu00004" & Exp_Date == "20-Sep-19" & Other_Treatment == "Whole water")[,23]

t.stat.vector.y <- filter(LE_H2O2.merged.size_frac, Row.names == "Otu00004" & Exp_Date == "20-Sep-19" & Other_Treatment == "100 um filtered")[,23]

t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "greater")
```
Yes, Cyanobium abundance is significantly higher in the whole water than in 100 um filtered water for some dates. We can conlcude that while it is generally not filtered out with 100um plankton mesh, some some Cyanobium cells were filtered onto the 100 um mesh in 24-Aug-19 and 20-Sep-19. The presence of Cyanobium on the colonies may have biological significance.  

Can we detect Lysinibacillus and the Bacillaceae OTUs in the mesocosm experiments, and are they associated with phytoplankton aggregates?  
```{r}
#Plot Lysinibacillus abundance:
filter(LE_H2O2.merged.size_frac_sum, Row.names == "Otu00209") %>%
  ggplot(aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.title = element_blank()) +
    ylab("Lysinibacillus Otu209 Abundance (reads/mL)")

#Plot Bacillaceae OTU 324 abundance:
filter(LE_H2O2.merged.size_frac_sum, Row.names == "Otu00375") %>%
  ggplot(aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.title = element_blank()) +
    ylab("Bacillaceae Otu375 Abundance (reads/mL)")

#Plot Bacillales OTU 1351 abundance:
filter(LE_H2O2.merged.size_frac_sum, Row.names == "Otu01608") %>%
  ggplot(aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.title = element_blank()) +
    ylab("Bacillales Otu1608 Abundance (reads/mL)")
```

The Lysinibacillus and Bacillaceae/Bacillales OTUs are not present in significant abundance in any of the mesocosms; they are probably contaminants. Do they show up in the PBS control samples?  
```{r}
##First, we need to create a dataframe containing the OTU abundances for the Extraction controls:
#Convert the control sample dataframe made earlier to long format:
MC_2019.merged.controls.long <- gather(MC_2019.merged.controls, Sample_ID, Perc_abund, 2:7)

#Add colony metadata:
MC_2019.merged.controls.long <- merge(MC_2019.merged.controls.long, MC_2019.metadata, by="Sample_ID", all.x = TRUE)
drop <- c("Sample_Type", "Amplification", "Enough_reads") #remove these columns, because they are no longer useful
MC_2019.merged.controls.long <- MC_2019.merged.controls.long[, !(names(MC_2019.merged.controls.long) %in% drop) ]
```

```{r}
#Plot abundance of each OTU in the controls:
MC_2019.merged.controls.long$Row.names = with(MC_2019.merged.controls.long, reorder(Row.names, Perc_abund, mean))

filter(MC_2019.merged.controls.long, Row.names == "Otu00209" | Row.names == "Otu00375" | Row.names == "Otu01608" | Row.names == "Otu00001") %>% 
    ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
      geom_jitter(size=0.3, alpha=0.7) +
      stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
      stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
      ggtitle("Full Range") +
      theme(plot.title = element_text(size=12),
            axis.title.x = element_blank(), 
            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
            axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
            axis.ticks.length = unit(-0.1, "cm"),
            axis.ticks = element_line(size = 0.2)) +
      ylab("Read Count")

filter(MC_2019.merged.controls.long, Row.names == "Otu00209" | Row.names == "Otu00375" | Row.names == "Otu01608" | Row.names == "Otu00001") %>% 
    ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
      geom_jitter(size=0.3, alpha=0.7) +
      stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
      stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
      ggtitle("Range 5-0 reads") +
      theme(plot.title = element_text(size=12),
            axis.title.x = element_blank(), 
            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
            axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
            axis.ticks.length = unit(-0.1, "cm"),
            axis.ticks = element_line(size = 0.2)) +
      coord_cartesian(ylim=c(0,5)) +
      ylab("Read Count")
```
What is their absolute abundance in the colony samples?
```{r}
#Get a dataframe of absolute read counts for the 16-Sep-19 samples:
keep <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Date_Collected == "16-Sep-19" & MC_2019.metadata$Sample_Type != "PBS Blank"]
MC_2019.shared.16Sep <- MC_2019.shared[ , colnames(MC_2019.shared) %in% keep]
MC_2019.merged.16Sep <- merge(MC_2019.shared.16Sep, MC_2019.taxonomy, by="row.names", all = FALSE)
MC_2019.merged.16Sep$Taxonomy <- gsub("\\([0-9]*\\)", "", MC_2019.merged.16Sep$Taxonomy)
#Divide the taxonomy into separate columns for each rank for sorting
MC_2019.merged.16Sep <- separate(MC_2019.merged.16Sep, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Convert dataframe to long format:
MC_2019.merged.16Sep.long <- gather(MC_2019.merged.16Sep, Sample_ID, Read_count, 2:15)

#Add colony metadata:
MC_2019.merged.16Sep.long <- merge(MC_2019.merged.16Sep.long, MC_2019.metadata, by="Sample_ID", all.x = TRUE)
drop <- c("Sample_Type", "Amplification", "Enough_reads") #remove these columns, because they are no longer useful

#Plot with Microcystis
filter(MC_2019.merged.16Sep.long, Row.names == "Otu00209" | Row.names == "Otu00375" | Row.names == "Otu01608" | Row.names == "Otu00001") %>% 
    ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Read_count)) +
      geom_jitter(size=0.3, alpha=0.7) +
      stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
      stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
      ggtitle("Full Range") +
      theme(plot.title = element_text(size=12),
            axis.title.x = element_blank(),
            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
            axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
            axis.ticks.length = unit(-0.1, "cm"),
            axis.ticks = element_line(size = 0.2)) +
      ylab("Read Count")

#Plot without Microcystis
filter(MC_2019.merged.16Sep.long, Row.names == "Otu00209" | Row.names == "Otu00375" | Row.names == "Otu01608") %>%
    ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Read_count)) +
      geom_jitter(size=0.3, alpha=0.7) +
      stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
      stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
      ggtitle("Full Range") +
      theme(plot.title = element_text(size=12),
            axis.title.x = element_blank(),
            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
            axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
            axis.ticks.length = unit(-0.1, "cm"),
            axis.ticks = element_line(size = 0.2)) +
      ylab("Read Count")
```
These taxa are detectable in the control samples. The amount of Lysinibacillus reads in the controls is equal to or higher than the amount of Lysinibacillus reads in nearly half of the true colonies collected from the same date. The remaining samples have many more Lysinibacillus reads than the control.

Lets make sure that the other key indicators and abundant taxa are not contaminants:  
```{r}
#Create a vector of indicator taxa to loop through:
indicators <- row.names(indval$sign[indval$sign[,9] <= 0.05 | rowSums(indval$sign[,1:6]) >= 5, ])

#There is an error in the loop: no loop for break/next, jumping to top level
#I should try filtering out the dataframe separately first? Then loop through the data to plot, this way there is no plotting error for OTUs present in the colonies that are absent in the mesocosm data.
for (i in indicators) {
d <- filter(LE_H2O2.merged.size_frac_sum, Row.names == i)
  print(ggplot(d, aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle(as.character(i)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.title = element_blank()) +
    ylab("Abundance (reads/mL)"))
  flush.console()
}
```

Let's investiagte other OTUs that are of interest or suspected contaminants:
```{r}
#Create a vector of OTUs to loop through:
OTU_check <- c("Otu01416", "Otu00588")

#There is an error in the loop: no loop for break/next, jumping to top level
#I should try filtering out the dataframe separately first? Then loop through the data to plot, this way there is no plotting error for OTUs present in the colonies that are absent in the mesocosm data.
for (i in OTU_check) {
d <- filter(LE_H2O2.merged.size_frac_sum, Row.names == i)
  print(ggplot(d, aes(fill=Other_Treatment, y=mean, x=Exp_Date)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle(as.character(i)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.title = element_blank()) +
    ylab("Abundance (reads/mL)"))
  flush.console()
}
```


There are several Otus that are not significantly higher in the whole water vs 100 um filtered datasets or are absent in the mesocosm datasets (which were also extracted with a different extraction kit).

Are the taxonomic assignments of these Otus in pervious papers that have characterized kit-ome contaminants?
Format for list below is: Otu#; taxonomoy; contamination flag; NCBI blast hits (Megablast algorithm); whether a matching taxonomy is identified in kit-ome literature

Otu 209 Lysinibacillus (formerly Bacillus); not in mesocosms; Bacillus horikoshii strains (isolated from Salk Lake, puffer fish liver, marine sediments, plants); Yes (Sheik et al. 2018 and Salter et al. 2014)

Otu 343 Sphingomonas; Not in mesocosms; Sphingomonas from plants, yogurt, amphibian skin, etc.; Yes (Sheik et al. 2018 and Salter et al. 2014).

Otu 375 Bacillaceae; Not in mesocosms; Bacillus strains from glaciers, Red sea, salt lakes, etc. (99.6% ID); Yes (Sheik et al. 2018 and Salter et al. 2014).

Otu 595 Rhodobacteraceae; Not in mesocosms; Paracoccus strains from caves, soil, eukaryotic algae cultures, lake and marine sediments, permafrost (98.8%); Yes (Sheik et al. 2018 and Salter et al. 2014).

Otu 602 Bradyrhizobium; Not in mesocosms; Bradyrhizobium from plants, study of plate cultivation; Yes (Sheik et al. 2018 and Salter et al. 2014).

Otu 603 Rickettsiaceae; only present in one mesocoms at sig. abundance, not higher in whole water; Uncultured bacteria from an urban tropical catchment (99.6%, only one hit at 97% or greater); No

Otu 588 Ralstonia; Not in mesocosms, Ralstonia, Yes (Sheik et al. 2018 and Salter et al. 2014)

Otu 442 Silvanigrellaceae; Not significantly higher in whole water; Uncultured bacteria from aquifers, soils after volcanoe eruption, microbial mat in Antarctica, airborne microbes; No

Otu 694 Peredibacter; Not significantly higher in whole water and not consistant within an experiment date; Peredibacter from river flood-plains, soil, soil after volcanoe eruption, urban floodwater, wetlands; No

Otu 1608 Bacillales; Not in mesocosms; Bacillus species; Yes (Sheik et al. 2018 and Salter et al. 2014)

Otu 2806 Planococcaceae; Not in mesocosms; Bacillus and Lysinibacillus species; Yes (Sheik et al. 2018 and Salter et al. 2014)

Otu 4285 Rothia; Not in mesocosms; Rothia strains from human microbiome; No

Otu 4787 Cyanobium; Not in mesocosms and not very abundant in colonies; no 

Otu 4715 Abiotrophia; Not in mesocosms; uncultured sequences and Abiotriophia from human skin, soil, medical equipment, etc.; Yes (Sheik et al. 2018 and Salter et al. 2014)

Otu 1416 Staphylococcus; Not in mesocosms; Staphylococcus aureus and other Staph species from skin (100%); No but is likely carry over from mock community or skin.

Many of the suspected contaminants have been listed as known contaminants from extraction kits in the literature. However, 9 OTUs that were present in the mesocosms but not higher in whole water treatments were not listed as contaminants from extraction kits. Can we identify these OTUs in the control samples?
```{r}
#Plot abundance of each OTU in the controls:
filter(MC_2019.merged.controls.long, Row.names == "Otu00038" | Row.names == "Otu00059" | Row.names == "Otu00164" | Row.names == "Otu00171" | Row.names == "Otu00256" | Row.names == "Otu00402" | Row.names == "Otu00442" | Row.names == "Otu00603" | Row.names == "Otu00612" | Row.names == "Otu00694" | Row.names == "Otu04285") %>% 
    ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
      geom_jitter(size=0.3, alpha=0.7) +
      stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
      stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
      theme(plot.title = element_text(size=12),
            axis.title.x = element_blank(), 
            axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
            axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
            axis.ticks.length = unit(-0.1, "cm"),
            axis.ticks = element_line(size = 0.2)) +
      coord_cartesian(ylim=c(0,5)) +
      ylab("Read Count")
```
All but two of the OTUs are not in the PBS blanks, must be a real signal. (Note the small deviations from 0 in the points are caused by geom_jitter; in the data frame they have 0 counts, as reflected by the read dots and error bars ).  Otus 38 and 4285 are likely contaminants.

Remove the contaminant Otus (abundance is not significantly different from 0 in the mesocosms or have no differences in any filtered vs unfiltered mesocosms that have been identified as contaminants) from the colony dataset:  
```{r}
contaminants <- c("Otu00038", "Otu04285", "Otu00209", "Otu00343", "Otu00375", "Otu00595", "Otu00602", "Otu00588", "Otu01608", "Otu02806" , "Otu04715", "Otu01416")

#Reprocess the data frame, removing the contaminant OTUs:
#Load dataframes:
MC_2019.shared <- read.table("MC_2019.shared", header = TRUE, sep="\t")
MC_2019.taxonomy <- read.table("LE_H2O2.taxonomy", header = TRUE, sep="\t")
MC_2019.metadata <- read.table("MC_2019_metadata.txt", header= TRUE, sep="\t")

#Fix up OTU table and merge with taxonomy table:
rownames(MC_2019.shared) <- MC_2019.shared$Group
drop <- c("label", "numOtus", "Group")
MC_2019.shared <- MC_2019.shared[ , !(names(MC_2019.shared) %in% drop)]
MC_2019.shared <- t(MC_2019.shared) #transpose table so that OTUs are row names for merging
MC_2019.shared <- MC_2019.shared[rowSums(MC_2019.shared) != 0,] #Remove Otus that are absent from colony samples
rownames(MC_2019.taxonomy) <- MC_2019.taxonomy$OTU
drop <- "OTU"
MC_2019.taxonomy <- MC_2019.taxonomy[ , !(names(MC_2019.taxonomy) %in% drop)]
MC_2019.taxonomy <- MC_2019.taxonomy[ MC_2019.taxonomy$Size > 2, ] #Only keep OTUs that have more than 2 sequences
MC_2019.merged <- merge(MC_2019.shared, MC_2019.taxonomy, by="row.names", all = FALSE)
#Divide the taxonomy into separate columns for each rank for sorting
MC_2019.merged <- separate(MC_2019.merged, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Filter dataframe so that we save the control samples into a separate dataframe for later
keep <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Sample_Type == "PBS Blank" ] 
taxcols <- c("Row.names", "Size", "Domain", "Phylum", "Class", "Order", "Family", "Genus")
MC_2019.merged.controls <- MC_2019.merged[ , names(MC_2019.merged) %in% keep | names(MC_2019.merged) %in% taxcols]

#Remove samples that were controls and had a low number of reads:
drop <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Enough_reads == "No" ]
MC_2019.merged <- MC_2019.merged[ , !(names(MC_2019.merged) %in% drop)]
MC_2019.merged.mock <- MC_2019.merged[ , names(MC_2019.merged) == "MC2019_Mock" | names(MC_2019.merged) == "Row.names"]

#Remove the Mock Sample:
MC_2019.merged <- MC_2019.merged[ , names(MC_2019.merged) != "MC2019_Mock"]

#Calculate the observation frequency of each OTU, add to dataframe:
otu_nonzero_counts <- apply(MC_2019.merged[2:45], 1, function(y) sum(length(which(y > 0))))
otu_obs_freq <- ( otu_nonzero_counts / 44 ) * 100
MC_2019.merged$otu_obs_freq <- otu_obs_freq

#Remove any OTUs not present in the colony samples:
MC_2019.merged <- MC_2019.merged[ MC_2019.merged$otu_obs_freq !=0, ]
Num_otus_prefilter <- nrow(MC_2019.merged)

#Remove the OTUs flagged as rare:
MC_2019.merged <- MC_2019.merged[ !(MC_2019.merged$Row.names %in% rares_to_remove), ]

#Remove the OTUs flagged as contaminants:
MC_2019.merged <- MC_2019.merged[ !(MC_2019.merged$Row.names %in% contaminants), ]
Num_otus_postfilter <- nrow(MC_2019.merged)

#How many OTUs were removed?
Num_otus_prefilter - Num_otus_postfilter

#What is the percentage of OTUs kept?
(Num_otus_postfilter / Num_otus_prefilter) * 100

#Save a copy of the df with raw counts for rarefaction curves later:
Rarefaction_df <- MC_2019.merged

#Convert to percent abundance:
MC_2019.merged[, 2:45] <- apply(MC_2019.merged[,2:45], 2, function(x) (x / sum(x)) *100 )

#Add the maximum, mean, and median relative abundance for each OTU as a column in the dataframe (not including zeros):
MC_2019.merged.nonzeros <- MC_2019.merged
MC_2019.merged.nonzeros[MC_2019.merged.nonzeros == 0] <- NA
MC_2019.merged$Max_perc_abund <- apply(MC_2019.merged[2:45], 1, max)
MC_2019.merged$Median_perc_abund <- apply(MC_2019.merged.nonzeros[2:45], 1, median, na.rm=TRUE)
MC_2019.merged$Mean_perc_abund <- apply(MC_2019.merged.nonzeros[2:45], 1, mean, na.rm=TRUE)

#Convert dataframe to long format:
MC_2019.merged.long <- gather(MC_2019.merged, Sample_ID, Perc_abund, 2:45)

#Add colony metadata:
MC_2019.merged.long <- merge(MC_2019.merged.long, MC_2019.metadata, by="Sample_ID", all.x = TRUE)
drop <- c("Sample_Type", "Amplification", "Enough_reads") #remove these columns, because they are no longer useful
MC_2019.merged.long <- MC_2019.merged.long[, !(names(MC_2019.merged.long) %in% drop) ]

#Fix up phylum names for aesthetic purposes:
MC_2019.merged.long$Phylum <- gsub("Acidobacteriota", "Acidobacteria", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Actinobacteriota", "Actinobacteria", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Bacteria_unclassified", "Unclassified Bacteria", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Bacteroidota", "Bacteroidetes", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Cloacimonadota", "Cloacimonetes", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Deinococcota", "Deinococcus", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Gemmatimonadota", "Gemmatimonadetes", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Nitrospirota", "Nitrospirae", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Planctomycetota", "Planctomycetes", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Spirochaetota", "Spirochaetes", MC_2019.merged.long$Phylum)
MC_2019.merged.long$Phylum <- gsub("Verrucomicrobiota", "Verrucomicrobia", MC_2019.merged.long$Phylum)
```
A total of 238 OTUs were removed as either rares or contaminants, keeping ~50% of the total OTUs.

Reanalysis of cleaned datatables  
----
How does removing the abundant contaminants impact clustering on the NMDS?  
```{r}
#Take our processed OTU table and remove the metadata columns:
#Columns 1-47 contain the info we need, the rest is metadata
MC_2019.matrix <- MC_2019.merged[MC_2019.merged$Genus != "Microcystis_PCC-7914",1:45] #Remove the Microcystis Otu
rownames(MC_2019.matrix) <- MC_2019.matrix$Row.names
MC_2019.matrix <- MC_2019.matrix[ , colnames(MC_2019.matrix) != "Row.names" ]
#Transpose the matrix so that samples are rows and OTUs are columns:
MC_2019.matrix <- t(MC_2019.matrix)

#Build the plot using the metaNMDS function in vegan with three axes:
set.seed(123)
NMDS_3 <- metaMDS(MC_2019.matrix, distance = "bray", k=3, autotransform = FALSE, try = 40, trymax = 100)
```

Check that the distance on the NMDS plot fairly represents the measured dissimilarity:  
```{r}
stressplot(NMDS_3, xlim=c(0,1))
```
Plot the ordination:
```{r}
#Extract the axis coordinates (the NMDS scores):
NMDS_3.scores <- as.data.frame(scores(NMDS_3))

#Add metadata:
NMDS_3.scores <- merge(NMDS_3.scores, MC_2019.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE)
rownames(NMDS_3.scores) <- NMDS_3.scores$Row.names
drop <- "Row.names"
NMDS_3.scores <- NMDS_3.scores[ , !(names(NMDS_3.scores) %in% drop) ]

#Plot axis 1 vs axis 2:
NMDS_3.1v2.plot <- ggplot(NMDS_3.scores, aes(x=NMDS1, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Oligotype, colour=Date_Collected)) +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 1")

#Plot axis 1 vs axis 3:
NMDS_3.1v3.plot <- ggplot(NMDS_3.scores, aes(x=NMDS3, y=NMDS1)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Oligotype, colour=Date_Collected)) +
  theme_classic() +
  theme(plot.title = element_text(hjust=1, size=10),
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 1") +
  xlab("NMDS 3")

#Plot axis 2 vs axis 3:
NMDS_3.2v3.plot <- ggplot(NMDS_3.scores, aes(x=NMDS3, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Oligotype, colour=Date_Collected)) +
  ggtitle("Stress = 0.12") +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "bottom",
    legend.box = "vertical",
    legend.text = element_text(size = 10), legend.title = element_text(size = 12)) +
  ylab("NMDS 2") +
  xlab("NMDS 3")

NDMS3 <- NMDS_3.1v2.plot + NMDS_3.2v3.plot + guide_area() + NMDS_3.1v3.plot + plot_layout(nrow = 2, guide="collect")
NDMS3 <- NDMS3 + labs(colour = "Sampling Date")
NDMS3
ggsave("NDMS3.pdf", NDMS3, width = 169, height = 180, units = "mm", dpi=600)
```
3D plot with date and morphotype:
```{r}
NMDS_3_3D_date_morph <- plot_ly(NMDS_3.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = ~Date_Collected, colors = c("coral", "gold", "blue", "aquamarine", "deeppink2", "green3"), symbol = ~Morphotype, symbols = c("circle", "diamond", "square"), showlegend = F) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
         annotations = list(text = "Stress = 0.12",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))
NMDS_3_3D_date_morph
```
**Legend:**  
**Date**  
blue = 22-Jul-19  
magenta = 5-Aug-19  
gold = 19-Aug-19  
aqua = 3-Sep-19  
green = 9-Sep-19  
coral = 16-Sep-19  

**Morphotype**  
circle = aeruginosa  
diamond = novacekii  
square = wesenbergii  

Plot with date and oligotype:
```{r}
NMDS_3_3D_date_oligo <- plot_ly(NMDS_3.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = ~Date_Collected, colors = c("coral", "gold", "blue", "aquamarine", "deeppink2", "green3"), symbol = ~Oligotype, symbols = c("circle", "square", "diamond", "cross"), showlegend = F) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
         annotations = list(text = "Stress = 0.12",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))
NMDS_3_3D_date_oligo
```
It looks like most of the separation is now driven by the samples that are dominated by Cyanobium.
**Legend:**  
**Date**  
blue = 22-Jul-19  
magenta = 5-Aug-19  
gold = 19-Aug-19  
aqua = 3-Sep-19  
green = 9-Sep-19  
coral = 16-Sep-19  

**Oligotype**  
circle = Type 8432  
diamond = Type 8437  
square = Type 1993-8432
cross - Type 8432-8437

Let's run the ANOSIM tests to check for significant differences in the mean dissimilarities between sampling dates and morphotype:  
```{r}
#ANOSIM for collection date:
ano_date <- anosim(MC_2019.matrix, NMDS_3.scores$Date_Collected, distance = "bray", permutations = 9999)
#ANOSIM for colony morphology:
ano_morph <- anosim(MC_2019.matrix, NMDS_3.scores$Morphotype, distance = "bray", permutations = 9999)
#ANOSIM for oligotype:
ano_oligo <- anosim(MC_2019.matrix, NMDS_3.scores$Oligotype, distance = "bray", permutations = 9999)

ano_date
ano_morph
ano_oligo
```
There is a significant separation in mean dissimilarity based on date and oligotype. The effect of oligotype is stronger than the effect of date. There are also significant differences based on colony morphology but the correlation is near zero.  

Let's see if the samples cluster into predictable groups based on date and oligotype using hierarchical clustering:  
```{r}
#Make a distance matrix:
MC_2019_all.dist <- vegdist(MC_2019.matrix, method="bray")
#Cluster using hierarchical clustering
MC_2019_all.clust <- hclust(MC_2019_all.dist, method = "average")
#Get three groups from the hclust dendogram
MC_2019_all.groups <- cutree(MC_2019_all.clust, k=3)

#Convert clustering results to a dendogram format:
MC_2019_all.dendro <- as.dendrogram(MC_2019_all.clust)
MC_2019_all.dendro <- color_branches(MC_2019_all.dendro, k=5, col = c("black"), groupLabels = TRUE)
MC_2019_all.dendro <- set(MC_2019_all.dendro, "labels_cex", 0.5)
#Plot
Colony_dendrog_plot <- plot(MC_2019_all.dendro, ylab = "Bray-Curtis Dissimilarity")
```

Let's see if the dates other than 9-Sep-19 are significantly different from each other after removing the high Cyanobium samples:  
```{r}
#Create a matrix without the 9-Sep-19 samples:
drop_df <- filter(MC_2019.metadata, MC_2019.metadata$Date_Collected == "9-Sep-19" | MC_2019.metadata$Sample_ID == "MC2019_0093")
Samples_to_drop <- drop_df$Sample_ID
rm(drop_df)
MC_2019.matrix.filtered <- MC_2019.matrix[ !(row.names(MC_2019.matrix) %in% as.vector(Samples_to_drop)), ]

#Make an NMDS ordination without the 9-Sep-19 samples:
#Build the plot using the metaNMDS function in vegan with three axes:
set.seed(321)
NMDS_3.filtered <- metaMDS(MC_2019.matrix.filtered, distance = "bray", k=3, autotransform = FALSE, try = 40, trymax = 100)
stressplot(NMDS_3.filtered, xlim=c(0,1))
```
3D plot:  
```{r}
#Extract the axis coordinates (the NMDS scores):
NMDS_3.filtered.scores <- as.data.frame(scores(NMDS_3.filtered))

#Add metadata:
NMDS_3.filtered.scores <- merge(NMDS_3.filtered.scores, MC_2019.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE)
rownames(NMDS_3.filtered.scores) <- NMDS_3.filtered.scores$Row.names
drop <- "Row.names"
NMDS_3.filtered.scores <- NMDS_3.filtered.scores[ , !(names(NMDS_3.filtered.scores) %in% drop) ]

NMDS_3_3D_filtered <- plot_ly(NMDS_3.filtered.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = ~Date_Collected, colors = c("coral", "gold", "blue", "aquamarine", "deeppink2", "green3"), symbol = ~Oligotype, symbols = c("circle", "square", "diamond", "cross"), showlegend = F) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
          legend = list(title = "Morphotype"),
         annotations = list(text = "Stress = 0.11",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))

#ANOSIM for date
ano_date <- anosim(MC_2019.matrix.filtered, NMDS_3.filtered.scores$Date_Collected, distance = "bray", permutations = 9999)
#ANOSIM for oligotype
ano_oligo <- anosim(MC_2019.matrix.filtered, NMDS_3.filtered.scores$Oligotype, distance = "bray", permutations = 9999)

NMDS_3_3D_filtered
```
**Date**  
blue = 22-Jul-19  
magenta = 5-Aug-19  
gold = 19-Aug-19  
aqua = 3-Sep-19  
coral = 16-Sep-19 

**Oligotype**  
circle = Type 8432  
diamond = Type 8437  
square = Type 1993-8432
cross - Type 8432-8437

```{r}
ano_date
ano_oligo
```
The separation by date is much weaker when the samples rich in Cyanobium are removed from the ordination, but the separation is still significant. We can conclude that for the most part, the mean colony community dissimilarity are not very different on each date, but there are some differences by date. The significance of oligotype also remains statistically significant, and has much more explanatory power than sample date.  

Identifying Key Taxa
---
What are the indicator taxa for each date?  
```{r}
#With the high Cyanobium samples:
indval <- multipatt(MC_2019.matrix, NMDS_3.scores$Date_Collected, func = "IndVal.g", control = how(nperm=999))
summary(indval, indvalcomp=TRUE)
```

Plot the mean abundance of all the indicator OTUs:
```{r}
#Get a list of Otus specific to each group:
indval.Otus.16Sep <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,1] == 1 & indval$sign[,9] < 0.05, ])
indval.Otus.19Aug <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,2] == 1 & indval$sign[,9] < 0.05, ])
indval.Otus.3Sep <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,4] == 1 & indval$sign[,9] < 0.05, ])
indval.Otus.5Aug <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,5] == 1 & indval$sign[,9] < 0.05, ])
indval.Otus.9Sep <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 1 & indval$sign[,6] == 1 & indval$sign[,9] < 0.05, ])

#Plot for 16-Sep-19:
indval_Otus_16Sep <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.16Sep & Date_Collected == "16-Sep-19")

indval_Otus_16Sep_plot <- ggplot(indval_Otus_16Sep, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 16-Sep-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
  coord_cartesian(ylim = c(0,15)) +  
  ylab("Percent abundance")

#Plot for 9-Sep-19:
indval_Otus_9Sep <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.9Sep & Date_Collected == "9-Sep-19")
                             
indval_Otus_9Sep_plot <-ggplot(indval_Otus_9Sep, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 9-Sep-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,80)) +
    ylab("Percent abundance")

#Plot for 19-Aug-19:
indval_Otus_19Aug <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.19Aug & Date_Collected == "19-Aug-19")

indval_Otus_19Aug_plot <-ggplot(indval_Otus_19Aug, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 19-Aug-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,1.5)) +
    ylab("Percent abundance")

#Plot for 3-Sep-19:
indval_Otus_3Sep <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.3Sep & Date_Collected == "3-Sep-19")

indval_Otus_3Sep_plot <- ggplot(indval_Otus_3Sep, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 3-Sep-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,5)) +
    ylab("Percent abundance")

#Plot for 5-Aug-19:
indval_Otus_5Aug <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.5Aug & Date_Collected == "5-Aug-19")

indval_Otus_5Aug_plot <-ggplot(indval_Otus_5Aug, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs 5-Aug-19") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    coord_cartesian(ylim = c(0,2)) +
    ylab("Percent abundance")

combined_single_indicators <- (indval_Otus_5Aug_plot | indval_Otus_19Aug_plot ) / (indval_Otus_3Sep_plot |  indval_Otus_9Sep_plot | indval_Otus_16Sep_plot)
combined_single_indicators
ggsave("combined_single_indicators.pdf", plot = combined_single_indicators, width = 35, height = 30, units = "cm", dpi=320)
```
There are indicators for all July-August samples, what are their mean abundances?
```{r}
#Get a list of Otus specific to each group:
indval.Otus.early <- row.names(indval$sign[ indval$sign[,2] == 1 & indval$sign[,3] == 1 & indval$sign[,5] == 1 & indval$sign[,9] <= 0.05, ])
early_dates <- c("22-Jul-19", "5-Aug-19", "19-Aug-19")

#Plot:
indval_Otus_early <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.early & Date_Collected %in% early_dates)
indval_Otus_early$Row.names = with(indval_Otus_early, reorder(Row.names, Perc_abund, mean))

July_Aug_Ind_plot <- ggplot(indval_Otus_early, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=1, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=1) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs for July and August") +
    theme(plot.title = element_text(size=10),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=10, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
  ylab("Percent abundance")
ggsave("July_Aug_Ind_plot.pdf", plot = July_Aug_Ind_plot, width = 15, height = 10, units = "cm", dpi=320)
July_Aug_Ind_plot
```
Are there any OTUs that are present on each date, and what are the abundances?  
```{r}
#Get a list of all Otus not specific to a group (p-value will be NA):
indval.Otus.all_dates <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) == 6, ] )
indval.Otus.all_dates <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.all_dates & Row.names != "Otu00001")
indval.Otus.all_dates$Row.names = with(indval.Otus.all_dates, reorder(Row.names, Perc_abund, mean))

#Plot:
indval.Otus.all_dates.full <- ggplot(indval.Otus.all_dates, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=1, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=1) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("OTUs present on every date (full range)") +
    theme(plot.title = element_text(size=10),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=10, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
  coord_cartesian(ylim=c(0,50)) +
  ylab("Percent abundance")

indval.Otus.all_dates.zoom <- ggplot(indval.Otus.all_dates, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=1, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=1) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("OTUs present on every date (0-10%)") +
    theme(plot.title = element_text(size=10),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=10, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
  coord_cartesian(ylim=c(0,10)) +
  ylab("Percent abundance")
  
OTU_all_date_plot <- indval.Otus.all_dates.full + indval.Otus.all_dates.zoom
ggsave("OTU_all_date_plot.pdf", plot = OTU_all_date_plot, width = 20, height = 10, units = "cm", dpi=320)
OTU_all_date_plot
```

What OTUs are present in 5+ dates?
```{r}
#Get a list of all Otus not specific to a group (p-value will be NA):
indval.Otus.5_dates <- row.names(indval$sign[ rowSums(indval$sign[,1:6]) >= 5, ] )
indval.Otus.5_dates <- filter(MC_2019.merged.long, Row.names %in% indval.Otus.5_dates & Row.names != "Otu00001")
indval.Otus.5_dates$Row.names = with(indval.Otus.5_dates, reorder(Row.names, Perc_abund, mean))

#Plot:
indval.Otus.5_dates.full <- ggplot(indval.Otus.5_dates, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=1, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=1) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("OTUs present on 5+ dates (full range)") +
    theme(plot.title = element_text(size=10),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=10, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
  coord_cartesian(ylim=c(0,50)) +
  ylab("Percent abundance")

indval.Otus.5_dates.zoom <- ggplot(indval.Otus.5_dates, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=1, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=1) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("OTUs present on 5+ dates (0-10%)") +
    theme(plot.title = element_text(size=10),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=10, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
  coord_cartesian(ylim=c(0,10)) +
  ylab("Percent abundance")
  
indval.Otus.5_dates.full + indval.Otus.5_dates.zoom  
```

Are there any indicator taxa for the oligotypes?  
```{r}
indval_oligo <- multipatt(MC_2019.matrix, NMDS_3.scores$Oligotype, func = "IndVal.g", control = how(nperm=999))
summary(indval_oligo, indvalcomp=TRUE)
```
What are the abundances of the indicator taxa for each Microcystis oligotype?  
```{r}
#Get a list of Otus specific to each Oligtoype:
indval_olig.Type2 <- row.names(indval_oligo$sign[ rowSums(indval_oligo$sign[,1:4]) == 1 & indval_oligo$sign[,2] == 1 & indval_oligo$sign[,7] < 0.05, ])
indval_olig.Type3 <- row.names(indval_oligo$sign[ rowSums(indval_oligo$sign[,1:4]) == 1 & indval_oligo$sign[,3] == 1 & indval_oligo$sign[,7] < 0.05, ])
indval_olig.Type4 <- row.names(indval_oligo$sign[ rowSums(indval_oligo$sign[,1:4]) == 1 & indval_oligo$sign[,4] == 1 & indval_oligo$sign[,7] < 0.05, ])

#Reorder the OTUs by mean percent abundance:
MC_2019.merged.long$Row.names = with(MC_2019.merged.long, reorder(Row.names, Perc_abund, mean))

#Plot of indcators for Type 2:
indval_Otus_Type2 <- filter(MC_2019.merged.long, Row.names %in% indval_olig.Type2 & Oligotype == "Oligotype 2")

indval_Otus_Type2_plot <-ggplot(indval_Otus_Type2, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs Type 2") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    ylab("Percent abundance")

#Plot of indcators for Tye 3:
indval_Otus_Type3 <- filter(MC_2019.merged.long, Row.names %in% indval_olig.Type3 & Oligotype == "Oligotype 3")

indval_Otus_Type3_plot <-ggplot(indval_Otus_Type3, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Indicator OTUs Type 3") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    ylab("Percent abundance")

combined_indval_oligo_plot <- indval_Otus_Type2_plot + indval_Otus_Type3_plot
combined_indval_oligo_plot
ggsave("combined_indval_oligo_plot.pdf", plot = combined_indval_oligo_plot, width = 32, height = 25, units = "cm", dpi=320)
```

Which OTUs are present on each oligotype?  
```{r}
#Get a list of Otus specific to each Oligtoype:
indval_olig.all <- row.names(indval_oligo$sign[ rowSums(indval_oligo$sign[,1:4]) == 4, ])

#Reorder the OTUs by mean percent abundance:
MC_2019.merged.long$Row.names = with(MC_2019.merged.long, reorder(Row.names, Perc_abund, mean))

#Plot:
indval_Otus_olig_all <- filter(MC_2019.merged.long, Row.names %in% indval_olig.all & Row.names != "Otu00002")

olig_all_Zoom_out <-ggplot(indval_Otus_olig_all, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Full Range") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    ylab("Percent abundance")


indval_Otus_olig_all_plot <- olig_all_Zoom_out
indval_Otus_olig_all_plot
ggsave("indval_Otus_olig_all_plot.pdf", plot = indval_Otus_olig_all_plot, width = 20, height = 16, units = "cm", dpi=320)
```
Which OTUs are present in all but Oligotype 4 (which has n=1):
```{r}
#Get a list of Otus specific to each Oligtoype:
indval_olig.3_types <- row.names(indval_oligo$sign[ rowSums(indval_oligo$sign[,1:3]) == 3, ])

#Reorder the OTUs by mean percent abundance:
MC_2019.merged.long$Row.names = with(MC_2019.merged.long, reorder(Row.names, Perc_abund, mean))

#Plot:
indval_Otus_olig_3types <- filter(MC_2019.merged.long, Row.names %in% indval_olig.3_types)

olig_3types_Zoom_out <-ggplot(indval_Otus_olig_3types, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
    geom_jitter(size=2, alpha=0.7) +
    stat_summary(fun.y=mean, geom="point", color="red", size=3) +
    stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.5, fun.args = list(mult = 1, conf.int = 0.95)) +
  ggtitle("Full Range") +
    theme(plot.title = element_text(size=15),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=15, margin = margin(t = 0, r = 15, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 12, color = "black", angle=60, hjust=1,  margin = margin(t = 19, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 12, color = "black", margin = margin(t = 0, r = 19, b = 0, l = 0)),
      axis.ticks.length = unit(-0.3, "cm"),
      axis.ticks = element_line(size = 0.5),
      legend.position = "none") +
    ylab("Percent abundance")


olig_3types_Zoom_out
#ggsave("indval_Otus_olig_all_plot.pdf", plot = indval_Otus_olig_all_plot, width = 20, height = 16, units = "cm", dpi=320)
```

Plot the top 20 OTUs based on mean abundance across all samples:  
```{r}
#Plot the percent abundance of each of the top 20 OTUs, based on mean abundance across all colonies:
MC_2019.merged.long$Row.names = with(MC_2019.merged.long, reorder(Row.names, Perc_abund, mean))
All_Otu_sorted <- unique(sort(MC_2019.merged.long$Row.names, decreasing=TRUE))[1:20]

MC2019_rank_abund_plot <- ggplot(MC_2019.merged.long, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
  geom_jitter(size=0.3, alpha=0.7) +
  stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2),
    legend.position = "none") +
  scale_x_discrete(limits = All_Otu_sorted) +
  coord_cartesian(ylim = c(0,100)) +
  ylab("Percent abundance")

MC2019_rank_abund_plot

#Replot but zoom the axis to better visualize the tail.
MC2019_rank_abund_plot_low <- ggplot(MC_2019.merged.long, aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
  geom_jitter(size=0.3, alpha=0.7) +
  stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2),
    legend.position = "none") +
  scale_x_discrete(limits = All_Otu_sorted) +
  coord_cartesian(ylim = c(0,10)) +
  ylab("Percent abundance")

MC2019_rank_abund_plot_low
ggsave("MC2019_rank_abund_plot.pdf", plot = MC2019_rank_abund_plot, width=18, height=10, units = "cm", dpi=320)
```
3 of the OTUs present on each date are in the top 20 most abundant OTUs based on mean percent abundance.

Let's plot mean abundance vs observation frequency for each OTU to get a sense of how conserved these taxa are in the Microcystis colonies:  
```{r}
phylum_colors <- c("moccasin",
"snow",
"yellow",
"red",
"blue",
"#b0005f",
"cyan",
"#29b83a",
"chartreuse",
"#ff6558",
"#0075ec",
"gray36",
"#cc9dff",
"#006c2b",
"#ff8dcd",
"olivedrab1",
"gray73",
"#018bd9",
"sienna4",
"blueviolet",
"#ffb166",
"gray0",
"#e2c283",
"navy",
"#ff8e94",
"gray88")

obs_freq_vs_abund_plot <- filter(MC_2019.merged.long, Sample_ID == "MC2019_0011" & Genus != "Microcystis_PCC-7914") %>% #This filtering step dereplicates the summary data. Needed for overplotting purposes
  ggplot(aes(x=otu_obs_freq, y=Max_perc_abund, size=Mean_perc_abund, fill=Phylum)) +
    geom_point(alpha=0.7, shape=21, color="black", stroke = 0.2) +
    scale_fill_manual(values = phylum_colors, name="           Phylum\n                or\nProteobacteria Class") +
    scale_size(range=c(0.5,10), breaks = c(0.1, 1, 5, 10, 15, 20, 25, 40), name="Mean percent abundance\n       (excluding zeros)") +
    theme_classic() +
    theme(
      axis.line.x = element_line(size=0.1),
      axis.line.y = element_line(size=0.1),
      axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_line(size=0.1),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.1),
      legend.text = element_text(size = 7),
      legend.direction = "horizontal", legend.position = "bottom", legend.box = "vertical") +
    coord_cartesian(ylim=c(0,100), xlim=c(0,100)) +
    xlab("Observation frequency (%)") +
    ylab("Maximum percent abundance")

ggsave("Otu_obs_freq_vs_abund.pdf", plot = obs_freq_vs_abund_plot, width = 169, height = 130, units = "mm", dpi=320)
obs_freq_vs_abund_plot
```
It looks like there are 4 Otus that are present in over 50% of Microcystis colonies with a max and mean percent abundance of 60-25% and 2-4% respectively. There's also some more occasionally occuring taxa (in 25-75% of colonies) that can reach higher percent abundances. Let's look at the abundance distribution of these OTUs across all the colonies:  
```{r}
MC_2019.merged.long$Row.names = with(MC_2019.merged.long, reorder(Row.names, otu_obs_freq, mean))

Frequent_OTUs <- filter(MC_2019.merged.long, otu_obs_freq >= 50 & Genus != "Microcystis_PCC-7914") %>%
  ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
  geom_jitter(size=0.3, alpha=0.7) +
  stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2),
    legend.position = "none") +
  coord_cartesian(ylim = c(0,60)) +
  ylab("Percent abundance")

Occasional_OTUs <- filter(MC_2019.merged.long, otu_obs_freq >= 25 & otu_obs_freq < 50 & Genus != "Microcystis_PCC-7914") %>%
  ggplot(aes(x=reorder(Row.names, desc(Row.names)), y=Perc_abund)) +
  geom_jitter(size=0.3, alpha=0.7) +
  stat_summary(fun.y=mean, geom="point", color="red", size=0.5) +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="red", size=0.2, fun.args = list(mult = 1, conf.int = 0.95)) +
  theme(axis.title.x = element_blank(), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.2),
    legend.position = "none") +
  coord_cartesian(ylim = c(0,25)) +
  ylab("Percent abundance")

Frequent_OTUs
Occasional_OTUs
```
Is the dissimilarity in the community composition between colonies explained by dissimilarities in environmental variation?
```{r}
#Create the dataframe needed for the Mantel test (we need to line up metadata and OTU abundances for each sample):
mantel_df <- t(MC_2019.merged[2:nrow(MC_2019.merged),1:45])
colnames(mantel_df) <- mantel_df[1,]
mantel_df <- mantel_df[2:45,]
mantel_df <- merge(mantel_df, MC_2019.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE, all.y = FALSE)

#Extract OTU abundance dataframe:
mantel_OTU_abund <- as.matrix(mantel_df[,2:235]) #The as.matrix will convert from factor to character (necessary step)
mantel_OTU_abund <- apply(mantel_OTU_abund, 2, as.numeric) #Need to convert the data to numeric from character.
#Get vectors of environmental parameters:
mantel_df_Temp <- mantel_df$Temp
mantel_df_SpCond <- mantel_df$SpCond
mantel_df_Part_MC <- mantel_df$Part_MC
mantel_df_O2 <- mantel_df$DissO2
mantel_df_PAR <- mantel_df$avgPAR_0.5m
mantel_df_TP <- mantel_df$TP
mantel_df_TDP <- mantel_df$TDP
mantel_df_SRP <- mantel_df$SRP
mantel_df_NH4 <- mantel_df$NH4
mantel_df_NO3 <- mantel_df$NO3_NO2
mantel_df_CDOM <- mantel_df$CDOM_400nm
mantel_df_CDOM_PAR <- select(mantel_df, avgPAR_0.5m, CDOM_400nm)
mantel_df_nuts <- select(mantel_df, TDP, SRP, NH4, NO3_NO2)
mantel_df_P <- select(mantel_df, TDP, SRP)
mantel_df_N <- select(mantel_df, NH4, NO3_NO2)
mantel_df_Area <- mantel_df$Area

#Convert the vectors and OTU abundance table into distance metrices:
#Used Bray-Curtis dissimilarity for OTU abundance and eulicdean distance for environmental parameters
dist.abund <- vegdist(mantel_OTU_abund, method = "bray")
dist.temp <- dist(mantel_df_Temp, method = "euclidean")
dist.SpCond <- dist(mantel_df_SpCond, method = "euclidean")
dist.Part_MC <- dist(mantel_df_Part_MC, method = "euclidean")
dist.O2 <- dist(mantel_df_O2, method = "euclidean")
dist.PAR <- dist(mantel_df_PAR, method = "euclidean")
dist.TP <- dist(mantel_df_TP, method = "euclidean")
dist.TDP <- dist(mantel_df_TDP, method = "euclidean")
dist.SRP <- dist(mantel_df_SRP, method = "euclidean")
dist.NH4 <- dist(mantel_df_NH4, method = "euclidean")
dist.NO3 <- dist(mantel_df_NO3, method = "euclidean")
dist.CDOM <- dist(mantel_df_CDOM, method = "euclidean")
dist.CDOM_PAR <- dist(mantel_df_CDOM_PAR, method = "euclidean")
dist.nuts <- dist(mantel_df_nuts, method = "euclidean")
dist.P <- dist(mantel_df_P, method = "euclidean")
dist.N <- dist(mantel_df_N, method = "euclidean")
dist.Area <- dist(mantel_df_Area, method = "euclidean")

#Run the mantel tests:
abund_v_temp <- mantel(dist.abund, dist.temp, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_SpCond <- mantel(dist.abund, dist.SpCond, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_Part_MC <- mantel(dist.abund, dist.Part_MC, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_O2 <- mantel(dist.abund, dist.O2, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_PAR <- mantel(dist.abund, dist.PAR, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_TP <- mantel(dist.abund, dist.TP, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_TDP <- mantel(dist.abund, dist.TDP, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_SRP <- mantel(dist.abund, dist.SRP, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_NH4 <- mantel(dist.abund, dist.NH4, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_NO3 <- mantel(dist.abund, dist.NO3, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_CDOM <- mantel(dist.abund, dist.CDOM, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_CDOM_PAR <- mantel(dist.abund, dist.CDOM_PAR, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_nuts <- mantel(dist.abund, dist.nuts, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_P <- mantel(dist.abund, dist.P, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_N <- mantel(dist.abund, dist.N, method = "spearman", permutations = 9999, na.rm = TRUE)
abund_v_Area <- mantel(dist.abund, dist.Area, method = "spearman", permutations = 9999, na.rm = TRUE)

abund_v_temp
abund_v_SpCond
abund_v_Part_MC
abund_v_O2
abund_v_PAR
abund_v_TP
abund_v_TDP
abund_v_SRP
abund_v_NH4
abund_v_NO3
abund_v_CDOM
abund_v_CDOM_PAR
abund_v_nuts
abund_v_P
abund_v_N
abund_v_Area
```
There are no strong correlations between the dissimilarity in environmental parameters and colony microbiomes. The strongest correlation is with CDOM, Temp, NO3, and PAR. Lets visualize what that data looks like:  
```{r}
#Vectorize the distance metrices for Bray Curtis dissimilarity, CDOM, and PAR and store into a new dataframe:
dist_df <- data.frame(as.vector(dist.abund), as.vector(dist.CDOM), as.vector(dist.PAR), as.vector(dist.CDOM_PAR), as.vector(dist.temp), as.vector(dist.NO3))
colnames(dist_df) <- c("Abund", "CDOM", "PAR", "CDOM_PAR", "Temp", "NO3")

#community dissimilarity vs CDOM dissimilarity:
CDOM_v_abund <- ggplot(dist_df, aes(x=CDOM, y=Abund)) +
  geom_point(size=2, alpha=0.7) +
  geom_smooth(method="lm", color="black") +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)),
        axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 9, color = "black", margin = margin(t = 5, r = 0, b = 0, l = 0)),
        axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
        axis.ticks.length = unit(-0.1, "cm"),
        axis.ticks = element_line(size = 0.2)) +
  coord_cartesian(ylim=c(0,1), xlim=c(0,1.5)) +
  xlab("Dissimilarity in CDOM (a400nm)") +
  ylab("Bray-Curtis Dissimilarity")

#community dissimilarity vs PAR dissimilarity:
PAR_v_abund <- ggplot(dist_df, aes(x=PAR, y=Abund)) +
  geom_point(size=2, alpha=0.7) +
  geom_smooth(method="lm", color="black") +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)),
        axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 9, color = "black", margin = margin(t = 5, r = 0, b = 0, l = 0)),
        axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
        axis.ticks.length = unit(-0.1, "cm"),
        axis.ticks = element_line(size = 0.2)) +
  coord_cartesian(ylim=c(0,1), xlim=c(0,100)) +
  xlab(expression("Dissimilarity in PAR ("*mu*"E/cm"^2*"/s)")) +
  ylab("Bray-Curtis Dissimilarity")

#community dissimilarity vs CDOM+PAR dissimilarity:
CDOM_Par_v_abund <- ggplot(dist_df, aes(x=CDOM_PAR, y=Abund)) +
  geom_point(size=2, alpha=0.7) +
  geom_smooth(method="lm", color="black") +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)),
        axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 9, color = "black", margin = margin(t = 5, r = 0, b = 0, l = 0)),
        axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
        axis.ticks.length = unit(-0.1, "cm"),
        axis.ticks = element_line(size = 0.2)) +
  coord_cartesian(ylim=c(0,1), xlim=c(0,100)) +
  xlab(expression("Dissimilarity in PAR and CDOM")) +
  ylab("Bray-Curtis Dissimilarity")

PAR_CDOM_beta_diversity_correlations <- CDOM_v_abund + PAR_v_abund
PAR_CDOM_beta_diversity_correlations
ggsave("PAR_CDOM_beta_diversity_correlations.pdf", plot = PAR_CDOM_beta_diversity_correlations, width = 180, height = 100, units = "mm", dpi=320)
```
```{r}
#Now plot relationship with temperature and nitrate:
Temp_v_abund <- ggplot(dist_df, aes(x=Temp, y=Abund)) +
  geom_point(size=2, alpha=0.7) +
  geom_smooth(method="lm", color="black") +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)),
        axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 9, color = "black", margin = margin(t = 5, r = 0, b = 0, l = 0)),
        axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
        axis.ticks.length = unit(-0.1, "cm"),
        axis.ticks = element_line(size = 0.2)) +
  coord_cartesian(ylim=c(0,1)) +
  xlab(expression("Dissimilarity in Temperature")) +
  ylab("Bray-Curtis Dissimilarity")

NO3_v_abund <- ggplot(dist_df, aes(x=NO3, y=Abund)) +
  geom_point(size=2, alpha=0.7) +
  geom_smooth(method="lm", color="black") +
  theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)),
        axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size = 9, color = "black", margin = margin(t = 5, r = 0, b = 0, l = 0)),
        axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
        axis.ticks.length = unit(-0.1, "cm"),
        axis.ticks = element_line(size = 0.2)) +
  coord_cartesian(ylim=c(0,1)) +
  xlab(expression("Dissimilarity in Nitrate Concentration (mg/L N)")) +
  ylab("Bray-Curtis Dissimilarity")

Temp_Nitrate_beta_diversity_correlations <- Temp_v_abund + NO3_v_abund
Temp_Nitrate_beta_diversity_correlations
ggsave("Temp_Nitrate_beta_diversity_correlations.pdf", plot = Temp_Nitrate_beta_diversity_correlations, width = 180, height = 100, units = "mm", dpi=320)
```

```{r}
cor(dist_df$CDOM, dist_df$Abund, method="pearson")
linearMod_CDOM <- lm(Abund ~ CDOM, data=dist_df)
summary(linearMod_CDOM)

cor(dist_df$PAR, dist_df$Abund, method="pearson")
linearMod_PAR <- lm(Abund ~ PAR, data=dist_df)
summary(linearMod_PAR)

cor(dist_df$Temp, dist_df$Abund, method="pearson")
linearMod_Temp <- lm(Abund ~ Temp, data=dist_df)
summary(linearMod_Temp)

cor(dist_df$NO3, dist_df$Abund, method="pearson", use = "complete.obs")
linearMod_NO3 <- lm(Abund ~ NO3, data=dist_df)
summary(linearMod_NO3)
```
There's a very weak relationship but still a lot of spread in dissimilarity that is unrelated to any of the highly correlated environmental parameters.  

Let's look at how alpha diversity changes between colonies of specific sampling dates and oligotypes:  
Edit the counts dataframe to fit the required format for vegan:
```{r}
#Remove Microcystis OTUs:
Rarefaction_df <- Rarefaction_df[ Rarefaction_df$Genus != "Microcystis_PCC-7914", ]
#Remove metadata columns:
drop <- c("Size", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "otu_obs_freq")
Rarefaction_df <- Rarefaction_df[ , !(names(Rarefaction_df) %in% drop)]
#Remove Row names
rownames(Rarefaction_df) <- Rarefaction_df$Row.names
Rarefaction_df <- Rarefaction_df[ , colnames(Rarefaction_df) != "Row.names" ]
#Transpose the matrix so that samples are rows and OTUs are columns:
Rarefaction_df <- t(Rarefaction_df)
```

Draw a rarefaction curve for the samples before we assess alpha diversity:
```{r}
rarecurve <- rarecurve(Rarefaction_df, step = 5, xlab = "Sample Size", ylab = "Species", label = FALSE, col = c("green3", "green3", "green3", "green3", "blue", "blue", "blue", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "deeppink3", "gold", "gold", "gold", "gold", "gold", "gold", "aquamarine", "aquamarine", "aquamarine", "aquamarine", "aquamarine", "aquamarine", "aquamarine", "coral", "coral", "coral", "coral", "coral", "coral", "coral", "coral", "coral", "coral", "coral", "coral"))
```
Estimate species richness by extrapolating the rarefaction curves:
```{r}
est_richness <- estimateR(Rarefaction_df)
est_richness <- t(est_richness)
```

Let's calculate species richness for each colony and the mean richness for "early" and "late samples:  
```{r}
Shannon <- diversity(MC_2019.matrix)
Simpson <- diversity(MC_2019.matrix, "simpson")
Richness <- specnumber(MC_2019.matrix)
Chao1 <- est_richness[,2]
Evenness <- Shannon/log(Richness)
MC_2019.merged.long <- merge(MC_2019.merged.long, Shannon, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
MC_2019.merged.long <- merge(MC_2019.merged.long, Simpson, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
MC_2019.merged.long <- merge(MC_2019.merged.long, Richness, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
MC_2019.merged.long <- merge(MC_2019.merged.long, Chao1, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
MC_2019.merged.long <- merge(MC_2019.merged.long, Evenness, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
colnames(MC_2019.merged.long)[39:43] <- c("Shannon", "Simpson", "Richness", "Chao1", "Evenness")
```

Calculate alpha diversity statistics for each group, run a T-test:  
```{r}
#Calculate mean, sd, se, and 95% CI for the diversity metrices above:
MC_2019.merged.long.shannon_sum <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Category) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Shannon))), sd=sd(as.numeric(as.character(Shannon)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.simpson_sum <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Category) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Simpson))), sd=sd(as.numeric(as.character(Simpson)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.richness_sum <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Category) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Richness))), sd=sd(as.numeric(as.character(Richness)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.chao_sum <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Category) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Chao1))), sd=sd(as.numeric(as.character(Chao1)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.evenness_sum <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Category) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Evenness))), sd=sd(as.numeric(as.character(Evenness)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

#T-test assuming unequal variance for Shannon:
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Early")[,39]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Late")[,39]
Shannon_t_test <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")

#T-test assuming unequal variance for Simpson:
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Early")[,40]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Late")[,40]
Simpson_t_test <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")

#T-test assuming unequal variance for Richness:
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Early")[,41]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Late")[,41]
Richness_t_test <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")

#T-test assuming unequal variance for Chao1:
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Early")[,42]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Late")[,42]
Chao1_t_test <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")

#T-test assuming unequal variance for Evenness:
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Early")[,43]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Late")[,43]
Evenness_t_test <-t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")

#Combine the dataframes:
MC_2019.merged.long.shannon_sum$Metric <- "Shannon"
MC_2019.merged.long.simpson_sum$Metric <- "Simpson"
MC_2019.merged.long.richness_sum$Metric <- "Richness"
MC_2019.merged.long.chao_sum$Metric <- "Chao1"
MC_2019.merged.long.evenness_sum$Metric <- "Evenness"
MC_2019.merged.long.alpha_sum <- rbind(MC_2019.merged.long.shannon_sum, MC_2019.merged.long.simpson_sum, MC_2019.merged.long.richness_sum, MC_2019.merged.long.chao_sum, MC_2019.merged.long.evenness_sum)
rm(MC_2019.merged.long.shannon_sum, MC_2019.merged.long.simpson_sum, MC_2019.merged.long.richness_sum, MC_2019.merged.long.evenness_sum, MC_2019.merged.long.chao_sum)

#Alpha Diversity plots
Richness_plot <- filter(MC_2019.merged.long.alpha_sum, Metric == "Richness") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Category)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle("p = 0.0001625") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,40)) +
    ylab("Species Richness")

Chao1_plot <- filter(MC_2019.merged.long.alpha_sum, Metric == "Chao1") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Category)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle("p = 0.281") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,50)) +
    ylab("Chao1")

shannon_plot <- filter(MC_2019.merged.long.alpha_sum, Metric == "Shannon") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Category)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle("p = 2.392e-08") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,3)) +
    ylab("Shannon Index")

simpson_plot <- filter(MC_2019.merged.long.alpha_sum, Metric == "Simpson") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Category)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle("p = 1.568e-06") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,1)) +
    ylab("Simpson Index")

evenness_plot <- filter(MC_2019.merged.long.alpha_sum, Metric == "Evenness") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Category)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    ggtitle("p = 0.0001067") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,0.7)) +
    ylab("Evenness")

combo_alpha_plot <- Richness_plot +  Chao1_plot + shannon_plot + simpson_plot + evenness_plot
combo_alpha_plot
ggsave("combo_alpha_plot.pdf", plot = combo_alpha_plot, width = 18, height = 12, units = "cm", dpi=320)
```
What is the distribution of Species Richness and Shannon Index?  
```{r}
#Richness histogram
Rich_hist <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  ggplot(aes(x=Richness)) +
    geom_histogram(breaks=c(0,5,10,20,30,40,50,60,70,80), color="black", fill="white") +
    scale_x_continuous(breaks = c(0,5,10,20,30,40,50,60,70,80)) +
    scale_y_continuous(breaks = seq(0,16, by = 2)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Species Richness") +
    ylab("Number of colonies")

#Chao1
Chao1_hist <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  ggplot(aes(x=Chao1)) +
    geom_histogram(breaks=c(0,5,10,20,30,40,50,60,70,80,90,100), color="black", fill="white") +
    scale_x_continuous(breaks = c(0,5,10,20,30,40,50,60,70,80,90,100)) +
    scale_y_continuous(breaks = seq(0,12, by = 2)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Chao1 Estimated Richness") +
    ylab("Number of colonies")

#Shannon histogram
Shannon_hist <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  ggplot(aes(x=Shannon)) +
  ggtitle("Single Colonies") +
    geom_histogram(breaks=c(seq(0,3, by = 0.2)), color="black", fill="white") +
    scale_x_continuous(breaks = seq(0,3, by = 0.2)) +
    scale_y_continuous(breaks = seq(0,14, by = 2)) +
    theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Shannon Index") +
    ylab("Number of colonies")

colony_alpha_hist <- Shannon_hist - Chao1_hist / Rich_hist
colony_alpha_hist
```

How does the alpha diversity change by date?
```{r}
#Calculate mean, sd, se, and 95% CI for the diversity metrices above for each date:
MC_2019.merged.long.shannon_by_date <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Collected) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Shannon))), sd=sd(as.numeric(as.character(Shannon)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.simpson_by_date <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Collected) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Simpson))), sd=sd(as.numeric(as.character(Simpson)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.richness_by_date <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Collected) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Richness))), sd=sd(as.numeric(as.character(Richness)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.chao_by_date <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Collected) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Chao1))), sd=sd(as.numeric(as.character(Chao1)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.evenness_by_date <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Collected) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Evenness))), sd=sd(as.numeric(as.character(Evenness)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

#Combine the dataframes:
MC_2019.merged.long.shannon_by_date$Metric <- "Shannon"
MC_2019.merged.long.simpson_by_date$Metric <- "Simpson"
MC_2019.merged.long.richness_by_date$Metric <- "Richness"
MC_2019.merged.long.chao_by_date$Metric <- "Chao1"
MC_2019.merged.long.evenness_by_date$Metric <- "Evenness"
MC_2019.merged.long.alpha_by_date <- rbind(MC_2019.merged.long.shannon_by_date, MC_2019.merged.long.simpson_by_date, MC_2019.merged.long.richness_by_date, MC_2019.merged.long.chao_by_date, MC_2019.merged.long.evenness_by_date)

rm(MC_2019.merged.long.shannon_by_date, MC_2019.merged.long.simpson_by_date, MC_2019.merged.long.richness_by_date, MC_2019.merged.long.evenness_by_date, MC_2019.merged.long.chao_by_date)

#Plot
Richness_plot_by_date <- filter(MC_2019.merged.long.alpha_by_date, Metric == "Richness") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Collected)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,45)) +
    scale_x_discrete(limits = c("22-Jul-19", "5-Aug-19", "19-Aug-19", "3-Sep-19", "9-Sep-19", "16-Sep-19")) +
    ylab("Species Richness")

Chao_plot_by_date <- filter(MC_2019.merged.long.alpha_by_date, Metric == "Chao1") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Collected)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,100)) +
    scale_x_discrete(limits = c("22-Jul-19", "5-Aug-19", "19-Aug-19", "3-Sep-19", "9-Sep-19", "16-Sep-19")) +
    ylab("Chao1 Est. Richness")

shannon_plot_by_date <- filter(MC_2019.merged.long.alpha_by_date, Metric == "Shannon") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Collected)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,3)) +
    scale_x_discrete(limits = c("22-Jul-19", "5-Aug-19", "19-Aug-19", "3-Sep-19", "9-Sep-19", "16-Sep-19")) +
    ylab("Shannon Index")

simpson_plot_by_date <- filter(MC_2019.merged.long.alpha_by_date, Metric == "Simpson") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Collected)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,1)) +
    scale_x_discrete(limits = c("22-Jul-19", "5-Aug-19", "19-Aug-19", "3-Sep-19", "9-Sep-19", "16-Sep-19")) +
    ylab("Simpson Index")

evenness_plot_by_date <- filter(MC_2019.merged.long.alpha_by_date, Metric == "Evenness") %>% ggplot(aes(fill=Metric, y=mean, x=Date_Collected)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,1)) +
    scale_x_discrete(limits = c("22-Jul-19", "5-Aug-19", "19-Aug-19", "3-Sep-19", "9-Sep-19", "16-Sep-19")) +
    ylab("Evenness")

combo_alpha_plot_by_date <- Richness_plot_by_date + Chao_plot_by_date + shannon_plot_by_date + simpson_plot_by_date + evenness_plot_by_date
combo_alpha_plot_by_date
ggsave("combo_alpha_plot_by_date.pdf", plot = combo_alpha_plot_by_date, width = 18, height = 12, units = "cm", dpi=320)
```
Is species richness or shannon index correlated with colony size?  
```{r}
MC_2019.merged.long.derep <- filter(MC_2019.merged.long, Row.names == "Otu00001")
cor(MC_2019.merged.long.derep$Shannon, MC_2019.merged.long.derep$Area)
linearMod_Shannon_Area <- lm(Shannon ~ Area, data=MC_2019.merged.long.derep)
summary(linearMod_Shannon_Area)

cor(MC_2019.merged.long.derep$Chao1, MC_2019.merged.long.derep$Area)
linearMod_Richness_Area <- lm(Chao1 ~ Area, data=MC_2019.merged.long.derep)
summary(linearMod_Richness_Area)
```
Plot the regressions:  
```{r}
Shannon_v_Area <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  ggplot(aes(x=Area, y=Shannon, color=Oligotype)) +
    geom_point() +
    geom_smooth(method=lm, color="navy", fill="lightsteelblue", se=TRUE,) +
    theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    coord_cartesian(ylim=c(0,3), xlim=c(0,3e5)) +
    xlab("Colony Area") +
    ylab("Shannon Index")

Richness_v_Area <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  ggplot(aes(x=Area, y=Chao1, color=Oligotype)) +
    geom_point() +
    geom_smooth(method=lm, color="navy", fill="lightsteelblue", se=TRUE,) +
    theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,200), xlim=c(0,3e5)) +
    xlab("Colony Area") +
    ylab("Chao1 Estimated Richness")

Diversity_vs_ColonySize <- Shannon_v_Area / Richness_v_Area
Diversity_vs_ColonySize
ggsave("Diversity_vs_ColonySize.pdf", plot = Diversity_vs_ColonySize, width = 18, height = 18, units = "cm", dpi=320)
```
There are no significant relationships between alpha diversity and colony size.  

Now plot alpha diversity grouped by Oligotype:  
```{r}
#Calculate mean, sd, se, and 95% CI for the diversity metrices above for each oligtoype:
MC_2019.merged.long.shannon_by_oligo <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Oligotype) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Shannon))), sd=sd(as.numeric(as.character(Shannon)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.simpson_by_oligo <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Oligotype) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Simpson))), sd=sd(as.numeric(as.character(Simpson)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.richness_by_oligo <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Oligotype) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Richness))), sd=sd(as.numeric(as.character(Richness)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.chao_by_oligo <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Oligotype) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Chao1))), sd=sd(as.numeric(as.character(Chao1)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

MC_2019.merged.long.evenness_by_oligo <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Oligotype) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Evenness))), sd=sd(as.numeric(as.character(Evenness)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

#Combine the dataframes:
MC_2019.merged.long.shannon_by_oligo$Metric <- "Shannon"
MC_2019.merged.long.simpson_by_oligo$Metric <- "Simpson"
MC_2019.merged.long.richness_by_oligo$Metric <- "Richness"
MC_2019.merged.long.chao_by_oligo$Metric <- "Chao1"
MC_2019.merged.long.evenness_by_oligo$Metric <- "Evenness"
MC_2019.merged.long.alpha_by_oligo <- rbind(MC_2019.merged.long.shannon_by_oligo, MC_2019.merged.long.simpson_by_oligo, MC_2019.merged.long.richness_by_oligo, MC_2019.merged.long.chao_by_oligo, MC_2019.merged.long.evenness_by_oligo)

rm(MC_2019.merged.long.shannon_by_oligo, MC_2019.merged.long.simpson_by_oligo, MC_2019.merged.long.richness_by_oligo, MC_2019.merged.long.evenness_by_oligo, MC_2019.merged.long.chao_by_oligo)

#Plot
Richness_plot_by_oligo <- filter(MC_2019.merged.long.alpha_by_oligo, Metric == "Richness") %>% ggplot(aes(fill=Metric, y=mean, x=Oligotype)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,45)) +
    ylab("Species Richness")

Chao_plot_by_oligo <- filter(MC_2019.merged.long.alpha_by_oligo, Metric == "Chao1") %>% ggplot(aes(fill=Metric, y=mean, x=Oligotype)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,120)) +
    ylab("Chao1 Est. Richness")

shannon_plot_by_oligo <- filter(MC_2019.merged.long.alpha_by_oligo, Metric == "Shannon") %>% ggplot(aes(fill=Metric, y=mean, x=Oligotype)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,3)) +
    ylab("Shannon Index")

simpson_plot_by_oligo <- filter(MC_2019.merged.long.alpha_by_oligo, Metric == "Simpson") %>% ggplot(aes(fill=Metric, y=mean, x=Oligotype)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,1)) +
    ylab("Simpson Index")

evenness_plot_by_oligo <- filter(MC_2019.merged.long.alpha_by_oligo, Metric == "Evenness") %>% ggplot(aes(fill=Metric, y=mean, x=Oligotype)) +
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci), width=0.2, position=position_dodge(0.9), color="black") +
    theme(plot.title = element_text(size=9),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2),
      legend.position = "none") +
    coord_cartesian(ylim=c(0,1)) +
    ylab("Evenness")

combo_alpha_plot_by_oligo <- Richness_plot_by_oligo + Chao_plot_by_oligo + shannon_plot_by_oligo + simpson_plot_by_oligo + evenness_plot_by_oligo
combo_alpha_plot_by_oligo
ggsave("combo_alpha_plot_by_oligo.pdf", plot = combo_alpha_plot_by_oligo, width = 18, height = 12, units = "cm", dpi=320)
```

Is average species richness of each oligotype significantly different?
```{r}
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 2")[,41]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 1")[,41]
Richness_t_test_1 <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Richness_t_test_1

t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 1")[,41]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 3")[,41]
Richness_t_test_2 <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Richness_t_test_2

t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 2")[,42]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 1")[,42]
Chao_t_test_1 <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Chao_t_test_1

t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 1")[,42]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 3")[,42]
Chao_t_test_2 <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Chao_t_test_2
```

Is average shannon index of each oligotype significantly different?
```{r}
t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 2")[,39]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 1")[,39]
Shannon_t_test_1 <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Shannon_t_test_1

t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 1")[,39]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Oligotype == "Oligotype 3")[,39]
Shannon_t_test_2 <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Shannon_t_test_2
```

What is the average relative abundance of Microcystis in early vs late samples?  
```{r}
MC_2019.merged.long.Microcystis_sum <- filter(MC_2019.merged.long, Row.names == "Otu00001") %>%
  group_by(Date_Category) %>%
  summarise(n=n(), mean=mean(as.numeric(as.character(Perc_abund))), sd=sd(as.numeric(as.character(Perc_abund)))) %>%
 mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

t.stat.vector.x <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Early")[,14]
t.stat.vector.y <- filter(MC_2019.merged.long, Row.names == "Otu00001" & Date_Category == "Late")[,14]
Microcystis_t_test <- t.test(t.stat.vector.x, t.stat.vector.y, paired = FALSE, alternative = "two.sided")
Microcystis_t_test
```

How distinct are the Non-Microcystis communities in the colonies from whole and 100 um filtered Lake Erie water?  
```{r}
#Import a combined data frame of all the field and colony samples:
contaminants <- c("Otu00038", "Otu04285", "Otu00209", "Otu00343", "Otu00375", "Otu00595", "Otu00602", "Otu00588", "Otu01608", "Otu02806" , "Otu04715", "Otu01416")

#Reprocess the data frame, removing the contaminant OTUs:
#Load dataframes:
LE_H2O2_2019_all.shared <- read.table("LE_H2O2_all.shared", header = TRUE, sep="\t")
LE_H2O2_2019_all.taxonomy <- read.table("LE_H2O2.taxonomy", header = TRUE, sep="\t")
LE_H2O2_2019_all.metadata <- read.table("LE_H2O2_all_metadata.txt", header= TRUE, sep="\t")

#Fix up OTU table and merge with taxonomy table:
rownames(LE_H2O2_2019_all.shared) <- LE_H2O2_2019_all.shared$Group
drop <- c("label", "numOtus", "Group")
LE_H2O2_2019_all.shared <- LE_H2O2_2019_all.shared[ , !(names(LE_H2O2_2019_all.shared) %in% drop)]
LE_H2O2_2019_all.shared <- t(LE_H2O2_2019_all.shared) #transpose table so that OTUs are row names for merging
LE_H2O2_2019_all.shared <- LE_H2O2_2019_all.shared[rowSums(LE_H2O2_2019_all.shared) != 0,] #Remove empty OTUs
rownames(LE_H2O2_2019_all.taxonomy) <- LE_H2O2_2019_all.taxonomy$OTU
drop <- "OTU"
LE_H2O2_2019_all.taxonomy <- LE_H2O2_2019_all.taxonomy[ , !(names(LE_H2O2_2019_all.taxonomy) %in% drop)]
LE_H2O2_2019_all.taxonomy <- LE_H2O2_2019_all.taxonomy[ LE_H2O2_2019_all.taxonomy$Size > 2, ] #Only keep OTUs that have more than 2 sequences
LE_H2O2_2019_all.merged <- merge(LE_H2O2_2019_all.shared, LE_H2O2_2019_all.taxonomy, by="row.names", all = FALSE)
#Divide the taxonomy into separate columns for each rank for sorting
LE_H2O2_2019_all.merged <- separate(LE_H2O2_2019_all.merged, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Remove samples that were controls and had a low number of reads:
drop <- LE_H2O2_2019_all.metadata$Sample_ID[ LE_H2O2_2019_all.metadata$Enough_reads == "No" ]
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[ , !(names(LE_H2O2_2019_all.merged) %in% drop)]

#Remove the Mock Sample:
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[ , names(LE_H2O2_2019_all.merged) != "MC2019_Mock"]

#Remove the cross-contaminated samples:
drop <- c("E2019_1152_1", "E2019_1177_1")
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[ , !(names(LE_H2O2_2019_all.merged) %in% drop)]

#Calculate the observation frequency of each OTU, add to dataframe:
otu_nonzero_counts <- apply(LE_H2O2_2019_all.merged[2:358], 1, function(y) sum(length(which(y > 0))))
otu_obs_freq <- ( otu_nonzero_counts / 357 ) * 100
LE_H2O2_2019_all.merged$otu_obs_freq <- otu_obs_freq

#Remove any OTUs with no sequences:
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[ LE_H2O2_2019_all.merged$otu_obs_freq != 0, ]
Num_otus_prefilter <- nrow(LE_H2O2_2019_all.merged)

#Remove the OTUs flagged as rare:
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[ !(LE_H2O2_2019_all.merged$Row.names %in% rares_to_remove), ]

#Remove the OTUs flagged as contaminants:
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[ !(LE_H2O2_2019_all.merged$Row.names %in% contaminants), ]
Num_otus_postfilter <- nrow(LE_H2O2_2019_all.merged)

#Remove OTUs from the internal standard:
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[LE_H2O2_2019_all.merged$Genus != "Thermus", ]

#How many OTUs were removed?
Num_otus_prefilter - Num_otus_postfilter

#What is the percentage of OTUs kept?
(Num_otus_postfilter / Num_otus_prefilter) * 100

#Save dataframe with raw counts for Chao1 calculations later:
Rarefaction_df <- LE_H2O2_2019_all.merged

#Create a datatable of Microcystis relative abundance to use for calculations later:
LE_H2O2_2019_all.Microcystis_abund <- LE_H2O2_2019_all.merged
LE_H2O2_2019_all.Microcystis_abund[,2:358] <- apply(LE_H2O2_2019_all.Microcystis_abund[,2:358], 2, function(x) (x / sum(x)) *100 )
LE_H2O2_2019_all.Microcystis_abund <- LE_H2O2_2019_all.Microcystis_abund[LE_H2O2_2019_all.Microcystis_abund$Genus == "Microcystis_PCC-7914", ]
LE_H2O2_2019_all.Microcystis_abund <- as.data.frame(apply(LE_H2O2_2019_all.Microcystis_abund[,2:358], 2, sum))
LE_H2O2_2019_all.Microcystis_abund$Sample_ID <- row.names(LE_H2O2_2019_all.Microcystis_abund)
colnames(LE_H2O2_2019_all.Microcystis_abund)[1] <- "Microcystis_rel_abund"

#Remove Microcystis OTUs, in the original submitted document this was done after calculation of percent abundance:
LE_H2O2_2019_all.merged <- LE_H2O2_2019_all.merged[LE_H2O2_2019_all.merged$Genus != "Microcystis_PCC-7914", ]

#Convert to percent abundance:
LE_H2O2_2019_all.merged[, 2:358] <- apply(LE_H2O2_2019_all.merged[,2:358], 2, function(x) (x / sum(x)) *100 )

#Add the maximum, mean, and median relative abundance for each OTU as a column in the dataframe (not including zeros):
LE_H2O2_2019_all.merged.nonzeros <- LE_H2O2_2019_all.merged
LE_H2O2_2019_all.merged.nonzeros[LE_H2O2_2019_all.merged.nonzeros == 0] <- NA
LE_H2O2_2019_all.merged$Max_perc_abund <- apply(LE_H2O2_2019_all.merged[2:358], 1, max)
LE_H2O2_2019_all.merged$Median_perc_abund <- apply(LE_H2O2_2019_all.merged.nonzeros[2:358], 1, median, na.rm=TRUE)
LE_H2O2_2019_all.merged$Mean_perc_abund <- apply(LE_H2O2_2019_all.merged.nonzeros[2:358], 1, mean, na.rm=TRUE)

#Convert dataframe to long format:
LE_H2O2_2019_all.merged.long <- gather(LE_H2O2_2019_all.merged, Sample_ID, Perc_abund, 2:358)

#Add metadata:
LE_H2O2_2019_all.merged.long <- merge(LE_H2O2_2019_all.merged.long, LE_H2O2_2019_all.metadata, by="Sample_ID", all.x = TRUE)
drop <- c("Amplification", "Enough_reads") #remove these columns, because they are no longer useful
LE_H2O2_2019_all.merged.long <- LE_H2O2_2019_all.merged.long[, !(names(LE_H2O2_2019_all.merged.long) %in% drop) ]
```

Calculate the Bray-Curtis dissimilarity and build the ordination plot:  
```{r}
#Take our processed OTU table and remove the metadata columns:
#Columns 1-358 contain the info we need, the rest is metadata
LE_H2O2_2019_all.matrix <- LE_H2O2_2019_all.merged[LE_H2O2_2019_all.merged$Genus != "Microcystis_PCC-7914",1:358] 
rownames(LE_H2O2_2019_all.matrix) <- LE_H2O2_2019_all.matrix$Row.names
LE_H2O2_2019_all.matrix <- LE_H2O2_2019_all.matrix[ , colnames(LE_H2O2_2019_all.matrix) != "Row.names" ]
#Transpose the matrix so that samples are rows and OTUs are columns:
LE_H2O2_2019_all.matrix <- t(LE_H2O2_2019_all.matrix)

#Build the plot using the metaNMDS function in vegan with three axes:
set.seed(419)
NMDS_3_all <- metaMDS(LE_H2O2_2019_all.matrix, distance = "bray", k=3, autotransform = FALSE, try = 40, trymax = 5000, maxit = 500, parallel = 4)
```
Let's try an ordination with only two axes:
```{r}
set.seed(444)
NMDS_2_all <- metaMDS(LE_H2O2_2019_all.matrix, distance = "bray", k=2, autotransform = FALSE, try = 40, trymax = 5000, maxit = 500, parallel = 4)
```

Test if distance on plot represents true dissimilarities:
```{r}
stressplot(NMDS_3_all, xlim=c(0,1))
```
As dissimilarity increases, so too does ordination distance. The stress is lower with three axes.

Cluster the samples using hierachical cluster analysis:
```{r}
#Make a distance matrix:
LE_H2O2_2019_all.dist <- vegdist(LE_H2O2_2019_all.matrix, method="bray")
#Cluster using hierarchical clustering
LE_H2O2_2019_all.clust <- hclust(LE_H2O2_2019_all.dist, method = "average")
#Get tree groups from the hclust dendogram
LE_H2O2_2019_all.groups <- cutree(LE_H2O2_2019_all.clust, k=6)
#merge groups with metadata to check sample type membership:
LE_H2O2_2019_all.groups.meta <- merge(LE_H2O2_2019_all.groups, LE_H2O2_2019_all.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE, all.y=FALSE)
```

Plot the results of the clustering:
```{r}
#Convert clustering results to a dendogram format:
LE_H2O2_2019_all.dendro <- as.dendrogram(LE_H2O2_2019_all.clust)
#Color the branches based on clustering.
LE_H2O2_2019_all.dendro <- color_branches(LE_H2O2_2019_all.dendro, k=6, col = c("red", "purple", "blue", "orange", "black", "cyan"), groupLabels = TRUE)
#Change the label size.
LE_H2O2_2019_all.dendro <- set(LE_H2O2_2019_all.dendro, "labels_cex", 0.5)
#Make the branches with
#Plot
plot(LE_H2O2_2019_all.dendro, ylab = "Bray-Curtis Dissimilarity")
```

Plot the NDMS ordination overlayed with the clusters:  
```{r}
#Extract the axis coordinates (the NMDS scores):
NMDS_3_all.scores <- as.data.frame(scores(NMDS_3_all))

#Add metadata:
NMDS_3_all.scores <- merge(NMDS_3_all.scores, LE_H2O2_2019_all.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE)
rownames(NMDS_3_all.scores) <- NMDS_3_all.scores$Row.names
drop <- "Row.names"
NMDS_3_all.scores <- NMDS_3_all.scores[ , !(names(NMDS_3_all.scores) %in% drop) ]

#Add cluster information:
LE_H2O2_2019_all.groups <- as.data.frame(LE_H2O2_2019_all.groups)
colnames(LE_H2O2_2019_all.groups) <- "Cluster"
NMDS_3_all.scores <- merge(NMDS_3_all.scores, LE_H2O2_2019_all.groups, by = "row.names")

#Plot
NMDS_3_3D_with_field <- plot_ly(NMDS_3_all.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = as.character(NMDS_3_all.scores$Cluster), colors = c("purple", "red", "blue", "orange", "black", "cyan"), symbol = as.character(NMDS_3_all.scores$Sample_Type), symbols=c('circle','diamond','square', "cross"), showlegend = F) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
         annotations = list(text = "Stress = 0.10",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))
NMDS_3_3D_with_field
```
ANOSIM test for sample type:  
```{r}
ano_sample_type <- anosim(LE_H2O2_2019_all.matrix, NMDS_3_all.scores$Sample_Type, distance = "bray", permutations = 9999)
ano_cluster <- anosim(LE_H2O2_2019_all.matrix, NMDS_3_all.scores$Cluster, distance = "bray", permutations = 9999)

ano_sample_type
ano_cluster
```
The statistic shows significant difference in ranked dissimilarities between groups.

Make a 2D plot for the paper:
```{r}
#Make the sample type entry prettier:
NMDS_3_all.scores$Sample_Type <- gsub("100um ret", expression("100 \U00B5m Retentate"),
                                      NMDS_3_all.scores$Sample_Type)
NMDS_3_all.scores$Sample_Type <- gsub("filtered", expression("105 \U00B5m Filtered"),
                                      NMDS_3_all.scores$Sample_Type)

#Plot axis 1 vs axis 2:
NMDS_3_all.1v2.plot <- ggplot(NMDS_3_all.scores, aes(x=NMDS1, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Sample_Type, colour=as.factor(Cluster))) +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 1")

#Plot axis 1 vs axis 3:
NMDS_3_all.1v3.plot <- ggplot(NMDS_3_all.scores, aes(x=NMDS3, y=NMDS1)) +
  geom_point(size = 3, alpha=0.6, aes(shape=Sample_Type, colour=as.factor(Cluster))) +
  theme_classic() +
  theme(plot.title = element_text(hjust=1, size=10),
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 1") +
  xlab("NMDS 3")

#Plot axis 2 vs axis 3:
NMDS_3_all.2v3.plot <- ggplot(NMDS_3_all.scores, aes(x=NMDS3, y=NMDS2)) +
  geom_point(size = 3, alpha=0.6, aes(shape=Sample_Type, colour=as.factor(Cluster))) +
  ggtitle("Stress = 0.10") +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "right",
    legend.box = "horizontal",
    legend.text = element_text(size = 10), legend.title = element_text(size = 12)) +
  ylab("NMDS 2") +
  xlab("NMDS 3")

NDMS3_with_field <- NMDS_3_all.1v2.plot + NMDS_3_all.2v3.plot + guide_area() + NMDS_3_all.1v3.plot + plot_layout(nrow=2, guides = "collect")
NDMS3_with_field
ggsave("NDMS3_with_field.pdf", NDMS3_with_field, width = 16.9, height = 18, units = "cm", dpi=600)
```
The >100 um bacterial communities do not separate as cleanly as before the correction. They are separated between clusters 1, 3, 6. These are labelled as A, C, and F in the published correction. Is the ordination of these clusters correlated with Microcystis relative abundance?

```{r}
#Add the Microcystis relative abundance to the metadata:
NMDS_3_all.scores <- merge(NMDS_3_all.scores, LE_H2O2_2019_all.Microcystis_abund, by.x = "Row.names", by.y = "Sample_ID", all.x = TRUE)

#Extract the 3 HC clusters that have 100 um Retentate samples:
#This step creates a dataframe of only retentate samples:
Retentate_Samples <- filter(NMDS_3_all.scores, Sample_Type == "100 m Retentate")
#Print out a list of all the HC clusters in that dataframe:
Cluster_vector <- unique(Retentate_Samples$Cluster)
#Go back to the original dataframe, and extract out all samples in those clusters that contain 100 um retentate samples:
NMDS_3_all.scores.Retentate_Clusters <- filter(NMDS_3_all.scores, Cluster %in% Cluster_vector)

#Is the cluster membership significantly correlated with Microcystis relative abundance?
#Take the abundance matrix used for NMDS, but only retain the samples in the clusters of interest:
LE_H2O2_2019_all.matrix.Retentate_Clusters <- filter(as.data.frame(LE_H2O2_2019_all.matrix), row.names(LE_H2O2_2019_all.matrix) %in% NMDS_3_all.scores.Retentate_Clusters$Row.names)

#Mantel test that the sample clustering in these clusters is correlated with Microcystis relative abundance
#Create the dataframe needed for the Mantel test (we need to line up metadata and OTU abundances for each sample):
Mantel_Microcystis_rel_abund_df <- merge(LE_H2O2_2019_all.matrix.Retentate_Clusters, NMDS_3_all.scores.Retentate_Clusters, by.x = "row.names", by.y = "Row.names", all = TRUE)

#Extract the Microcystis relative abundance vector:
Microcystis_rel_abund_vector <- Mantel_Microcystis_rel_abund$Microcystis_rel_abund

#Create a distance matrix for both non-Microcystis community composition and Microcystis relative abundance:
dist.abund.nonMC <- vegdist(LE_H2O2_2019_all.matrix.Retentate_Clusters, method = "bray")
dist.abund.MC <- dist(Microcystis_rel_abund_vector, method = "euclidean")

#Run the mantel test:
Mantel_Microcystis_rel_abund <- mantel(dist.abund.nonMC, dist.abund.MC, method = "spearman", permutations = 9999)
Mantel_Microcystis_rel_abund
```
The clustering of the samples in these three HC clusters is significantly correlated with Microcystis relative abundance.  
Make an NMDS plot colored by Microcystis relative abundance:  
```{r}
#Create a dataframe of the distances:
Ret_Cluster_MC_abund <- ggplot(NMDS_3_all.scores.Retentate_Clusters, aes(x=as.factor(Cluster), y=Microcystis_rel_abund)) +
  geom_boxplot(aes(fill=Sample_Type)) +
  stat_summary(fun.y=mean, geom="point", shape=18, size=5, color="gold", fill="gold") +
  stat_summary(fun.data=mean_cl_normal, geom="errorbar", color="gold", size=0.5, fun.args = list(mult = 1, conf.int = 0.95), aes(x=as.factor(Cluster), y=Microcystis_rel_abund)) +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "right",
    legend.box = "horizontal",
    legend.text = element_text(size = 10), legend.title = element_text(size = 12)) +
  xlab("HC Cluster") +
  ylab("Microcystis Relative Abundance")

Ret_Cluster_MC_abund
ggsave("Ret_Cluster_MC_abund.pdf", Ret_Cluster_MC_abund, width = 14, height = 12, units = "cm", dpi=600)
```
The mean Microcystis relative abundances look different for each cluster, but the 100 um retentate samples in Cluster 1 (A in the paper) can have just as high Microcystis relative abundance as those that clustered with the single colonies. Do the samples that cluster with single colonies have a higher mean Microcystis relative abundance?  

Run a Kruskal-Wallis Test to see if mean Microcystis relative abundance in the 100 um retentate samples is significantly different between the three HC Clusters under consideration:  
```{r}
Kruskal_MC_abund_v_Cluster <- kruskal.test(Microcystis_rel_abund ~ Cluster, data = Retentate_Samples)
Kruskal_MC_abund_v_Cluster
```
The three HC clusters have significantly different Microcystis relative abundance, but which ones are significantly different?
```{r}
Pairwise_Wilcox_MC_abund_v_Cluster <- pairwise.wilcox.test(Retentate_Samples$Microcystis_rel_abund, Retentate_Samples$Cluster,
                 p.adjust.method = "BH")
Pairwise_Wilcox_MC_abund_v_Cluster
```
The mean Microcystis relative abundance in the 100 um retentate samples in Clusters 1 and 3 (A and C in the paper) are not significantly different.  

Change the dendrogram previously made so that the colonies that the branches that point to the single colonies that clustered with 100 um retentate samples are dashed:
```{r}
#Get a vector of labels that I want to be dashed. These are single colonies that clustered with 100 um retentate samples in the 3rd hierarchical clustering group.
Select_entries.Retentate_Clusters <- filter(NMDS_3_all.scores.Retentate_Clusters, Cluster == "3" & Sample_Type == "Colony")

#Edit the dendrogram with the updated branch type:
LE_H2O2_2019_all.dendro <- set(LE_H2O2_2019_all.dendro, "by_labels_branches_lty", value = Select_entries.Retentate_Clusters$Row.names, TF_values = c(3,Inf))

#Now get a list of sample ids for all 100 um samples
Select_entries.Retentate_Clusters <- filter(NMDS_3_all.scores.Retentate_Clusters, Sample_Type == "100 m Retentate")

#Edit the dendrogram so that branches for 100 um retentate samples are thick:
LE_H2O2_2019_all.dendro <- set(LE_H2O2_2019_all.dendro, "by_labels_branches_lwd", value = Select_entries.Retentate_Clusters$Row.names, TF_values = c(2,Inf))

#Plot
plot(LE_H2O2_2019_all.dendro, ylab = "Bray-Curtis Dissimilarity")
```


Make an NMDS plot of just the whole water field samples:
```{r}
#Remove all the samples not from whole water:
drop <- LE_H2O2_2019_all.metadata$Sample_ID[LE_H2O2_2019_all.metadata$Sample_Type != "Whole Water"]
LE_H2O2.matrix <- LE_H2O2_2019_all.matrix[ !(row.names(LE_H2O2_2019_all.matrix) %in% drop), ]

#Make the NMDS ordination:
set.seed(763)
NMDS_3_field <- metaMDS(LE_H2O2.matrix, distance = "bray", k=3, autotransform = FALSE, try = 40, trymax = 2000)
```
Plot:
```{r}
#Extract the axis coordinates (the NMDS scores):
NMDS_3_field.scores <- as.data.frame(scores(NMDS_3_field))

#Add metadata:
NMDS_3_field.scores <- merge(NMDS_3_field.scores, LE_H2O2_2019_all.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE)
rownames(NMDS_3_field.scores) <- NMDS_3_field.scores$Row.names
drop <- "Row.names"
NMDS_3_field.scores <- NMDS_3_field.scores[ , !(names(NMDS_3_field.scores) %in% drop) ]

#Plot and color points by month
NMDS_3_3D_field_only <- plot_ly(NMDS_3_field.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = NMDS_3_field.scores$Month, symbol = NMDS_3_field.scores$Year, symbols=c('circle','diamond','square', 'cross'), showlegend = F) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
         annotations = list(text = "Stress = 0.091",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))
NMDS_3_3D_field_only
```
Create 2D NMDS plots for day of year:  
```{r}
#Create a date formatted month column so that the color gradient has the correct order.
NMDS_3_field.scores$Month_Date <- month(dmy(NMDS_3_field.scores$Date_Collected))

#Plot axis 1 vs axis 2:
NMDS_3_field.1v2.time.plot <- ggplot(NMDS_3_field.scores, aes(x=NMDS1, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=as.factor(Year), colour=Month_Date)) +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 1")

#Plot axis 1 vs axis 3:
NMDS_3_field.1v3.time.plot <- ggplot(NMDS_3_field.scores, aes(x=NMDS3, y=NMDS1)) +
  geom_point(size = 3, alpha=0.8, aes(shape=as.factor(Year), colour=Month_Date)) +
  theme_classic() +
  theme(plot.title = element_text(hjust=1, size=10),
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "right",
    legend.box = "vertical",
    legend.text = element_text(size = 12), legend.title = element_text(size = 12)) +
  ylab("NMDS 1") +
  xlab("NMDS 3")

#Plot axis 2 vs axis 3:
NMDS_3_field.2v3.time.plot <- ggplot(NMDS_3_field.scores, aes(x=NMDS3, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=as.factor(Year), colour=Month_Date)) +
  ggtitle("Stress = 0.09") +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 3")

NDMS3_field <- NMDS_3_field.1v2.time.plot + NMDS_3_field.2v3.time.plot + guide_area() + NMDS_3_field.1v3.time.plot + plot_layout(nrow = 2, guides = "collect")
NDMS3_field <- NDMS3_field + labs(shape = "Year")
NDMS3_field
ggsave("NDMS3_field.pdf", NDMS3_field, width = 16.9, height = 18, units = "cm", dpi=600)
```

```{r}
#Plot and color points by chlorophyll
NMDS_3_3D_field_chl <- plot_ly(NMDS_3_field.scores, x = ~NMDS1, y = ~NMDS2, z = ~NMDS3, color = log(NMDS_3_field.scores$Chl), symbol = NMDS_3_field.scores$Year, symbols=c('circle','diamond','square', 'cross'), showlegend = T) %>%
  add_markers(marker = list(line = list(color = "black", width = 0.5),
                            size = 6)) %>%
  layout(scene = list(xaxis = list(title = "NMDS1"),
                      yaxis = list(title = "NMDS2"),
                      zaxis = list(title = "NMDS3")),
         annotations = list(text = "Stress = 0.09",
                            showarrow = F,
                            align="left",
                            x=0.8,
                            y=0.9,
                            z=2.5))
NMDS_3_3D_field_chl
```
Create 2D NMDS plots colored by chlorophyll concentration:  
```{r}
#Plot axis 1 vs axis 2:
NMDS_3_field.1v2.Chl.plot <- ggplot(NMDS_3_field.scores, aes(x=NMDS1, y=NMDS2)) +
  geom_point(size = 3, alpha=0.6, aes(shape=as.factor(Year), colour=log(Chl))) +
  scale_color_gradient(low = "dodgerblue4", high = "green") +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 1")

#Plot axis 1 vs axis 3:
NMDS_3_field.1v3.Chl.plot <- ggplot(NMDS_3_field.scores, aes(x=NMDS3, y=NMDS1)) +
  geom_point(size = 3, alpha=0.6, aes(shape=as.factor(Year), colour=log(Chl))) +
  scale_color_gradient(low = "dodgerblue4", high = "green") +
  theme_classic() +
  theme(plot.title = element_text(hjust=1, size=10),
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "right",
    legend.box = "vertical",
    legend.text = element_text(size = 12), legend.title = element_text(size = 12)) +
  ylab("NMDS 1") +
  xlab("NMDS 3")

#Plot axis 2 vs axis 3:
NMDS_3_field.2v3.Chl.plot <- ggplot(NMDS_3_field.scores, aes(x=NMDS3, y=NMDS2)) +
  geom_point(size = 3, alpha=0.6, aes(shape=as.factor(Year), colour=log(Chl))) +
  scale_color_gradient(low = "dodgerblue4", high = "green") +
  ggtitle("Stress = 0.09") +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 3")

NDMS3_field_Chl <- NMDS_3_field.1v2.Chl.plot + NMDS_3_field.2v3.Chl.plot + guide_area() + NMDS_3_field.1v3.Chl.plot + plot_layout(nrow = 2, guides = "collect")
NDMS3_field_Chl <- NDMS3_field_Chl + labs(shape = "Year", colour = expression("ln Chlorophyll"*italic(" a")*" ("*mu*"g/L)"))
NDMS3_field_Chl
ggsave("NDMS3_field_Chl.pdf", NDMS3_field_Chl, width = 16.9, height = 18, units = "cm", dpi=600)
```
Run an anosim test for month:  
```{r}
anosim(LE_H2O2.matrix, NMDS_3_field.scores$Month, distance = "bray", permutations = 9999)
```
Run a mantel test for log transformed chlorophyll:  
```{r}
#Create the dataframe needed for the Mantel test (we need to line up metadata and OTU abundances for each sample):
WW_mantel_df <- merge(LE_H2O2.matrix, LE_H2O2_2019_all.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE, all.y = FALSE)

#WW_mantel_OTU_abund <- apply(mantel_OTU_abund, 2, as.numeric) #Need to convert the data to numeric from character.
#Extract chlorophyll vector:
WW_mantel_df_chl <- WW_mantel_df$Chl

#Create a distance matrix for both OTUs and chlorophyll concentration:
dist.abund.WW <- vegdist(LE_H2O2.matrix, method = "bray")
dist.chl.WW <- dist(WW_mantel_df_chl, method = "euclidean")

#Run the mantel test:
WW_abund_v_chl <- mantel(dist.abund.WW, dist.chl.WW, method = "spearman", permutations = 9999, na.rm = TRUE)

WW_abund_v_chl
```
Calculate the alpha diversity of the bulk samples to compare with the single Microcystis colonies
```{r}
#Remove Microcystis OTUs:
Rarefaction_df <- Rarefaction_df[ Rarefaction_df$Genus != "Microcystis_PCC-7914", ]
#Remove metadata columns:
drop <- c("Size", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "otu_obs_freq")
Rarefaction_df <- Rarefaction_df[ , !(names(Rarefaction_df) %in% drop)]
#Remove Row names
rownames(Rarefaction_df) <- Rarefaction_df$Row.names
Rarefaction_df <- Rarefaction_df[ , colnames(Rarefaction_df) != "Row.names" ]
#Transpose the matrix so that samples are rows and OTUs are columns:
Rarefaction_df <- t(Rarefaction_df)


LE_H2O2_2019_all_est_richness <- estimateR(Rarefaction_df)
LE_H2O2_2019_all_est_richness <- t(LE_H2O2_2019_all_est_richness)
LE_H2O2_2019_all_Chao1 <- LE_H2O2_2019_all_est_richness[,2]
LE_H2O2_2019_all_Shannon <- diversity(LE_H2O2_2019_all.matrix)
LE_H2O2_2019_all_Simpson <- diversity(LE_H2O2_2019_all.matrix, "simpson")
LE_H2O2_2019_all_Richness <- specnumber(LE_H2O2_2019_all.matrix)
LE_H2O2_2019_all_Evenness <- LE_H2O2_2019_all_Shannon/log(LE_H2O2_2019_all_Richness)
LE_H2O2_2019_all.merged.long <- merge(LE_H2O2_2019_all.merged.long, LE_H2O2_2019_all_Shannon, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
LE_H2O2_2019_all.merged.long <- merge(LE_H2O2_2019_all.merged.long, LE_H2O2_2019_all_Simpson, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
LE_H2O2_2019_all.merged.long <- merge(LE_H2O2_2019_all.merged.long, LE_H2O2_2019_all_Richness, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
LE_H2O2_2019_all.merged.long <- merge(LE_H2O2_2019_all.merged.long, LE_H2O2_2019_all_Chao1, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
LE_H2O2_2019_all.merged.long <- merge(LE_H2O2_2019_all.merged.long, LE_H2O2_2019_all_Evenness, all.x = TRUE, all.y = FALSE, by.x = "Sample_ID", by.y = "row.names")
colnames(LE_H2O2_2019_all.merged.long)[23:27] <- c("Shannon", "Simpson", "Richness", "Chao1", "Evenness")
rm(LE_H2O2_2019_all_Shannon)
rm(LE_H2O2_2019_all_Simpson)
rm(LE_H2O2_2019_all_Richness)
rm(LE_H2O2_2019_all_Evenness)
```

Make a histogram of species richness and shannon index for the bulk water samples.
```{r}
WW_Rich_hist <- filter(LE_H2O2_2019_all.merged.long, Row.names == "Otu00002" & Sample_Type == "Whole Water") %>%
  ggplot(aes(x=Richness)) +
    geom_histogram(breaks=seq(200,1600, by = 100), color="black", fill="white") +
    scale_x_continuous(breaks = seq(200,1600, by = 200)) +
    scale_y_continuous(breaks = seq(0,90, by = 10)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Species Richness") +
    ylab("Number of samples")

WW_chao_hist <- filter(LE_H2O2_2019_all.merged.long, Row.names == "Otu00002" & Sample_Type == "Whole Water") %>%
  ggplot(aes(x=Chao1)) +
    geom_histogram(breaks=seq(200,2600, by = 100), color="black", fill="white") +
    scale_x_continuous(breaks = seq(200,2600, by = 200)) +
    scale_y_continuous(breaks = seq(0,90, by = 10)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Chao1 Est. Richness") +
    ylab("Number of samples")

WW_Shannon_hist <- filter(LE_H2O2_2019_all.merged.long, Row.names == "Otu00002" & Sample_Type == "Whole Water") %>%
  ggplot(aes(x=Shannon)) +
    ggtitle("Whole Water Samples") +
    geom_histogram(breaks=c(seq(3,6, by = 0.5)), color="black", fill="white") +
    scale_y_continuous(breaks = seq(0,160, by = 40)) +
    theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Shannon Index") +
    ylab("Number of samples")

Filt_Rich_hist <- filter(LE_H2O2_2019_all.merged.long, Row.names == "Otu00002" & Sample_Type == "filtered") %>%
  ggplot(aes(x=Richness)) +
    geom_histogram(breaks=seq(200,1600, by = 100), color="black", fill="white") +
    scale_x_continuous(breaks = seq(200,1600, by = 200)) +
    scale_y_continuous(breaks = seq(0,8, by = 4)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Species Richness") +
    ylab("Number of samples")

Filt_chao_hist <- filter(LE_H2O2_2019_all.merged.long, Row.names == "Otu00002" & Sample_Type == "filtered") %>%
  ggplot(aes(x=Chao1)) +
    geom_histogram(breaks=seq(200,2600, by = 100), color="black", fill="white") +
    scale_x_continuous(breaks = seq(200,2600, by = 200)) +
    scale_y_continuous(breaks = seq(0,6, by = 2)) +
    theme(plot.title = element_text(size=12),
      axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Chao1 Est. Richness") +
    ylab("Number of samples")

Filt_Shannon_hist <- filter(LE_H2O2_2019_all.merged.long, Row.names == "Otu00002" & Sample_Type == "filtered") %>%
  ggplot(aes(x=Shannon)) +
    ggtitle("<100 um Samples") +
    geom_histogram(breaks=c(seq(3,6, by = 0.5)), color="black", fill="white") +
    scale_y_continuous(breaks = seq(0,20, by = 5)) +
    theme(axis.title.x = element_text(size=12, margin = margin(t = 14, r = 0, b = 0, l = 0)), 
      axis.title.y = element_text(size=12, margin = margin(t = 0, r = 14, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 9, color = "black", angle=60, hjust=1,  margin = margin(t = 5, r = 0, b = 0, l = 0)),
      axis.text.y = element_text(size = 9, color = "black", margin = margin(t = 0, r = 5, b = 0, l = 0)),
      axis.ticks.length = unit(-0.1, "cm"),
      axis.ticks = element_line(size = 0.2)) +
    xlab("Shannon Index") +
    ylab("Number of samples")


WW_alpha_hist <- WW_Shannon_hist - WW_chao_hist / WW_Rich_hist
Filt_alpha_hist <- Filt_Shannon_hist - Filt_chao_hist / Filt_Rich_hist
full_alpha_hist <- colony_alpha_hist / WW_alpha_hist / Filt_alpha_hist
full_alpha_hist
ggsave("full_alpha_hist_plot.pdf", plot = full_alpha_hist, width = 18, height = 33, units = "cm", dpi=320)
```
Let's look at how the abundances of indicator OTUs for oligotype and multiple dates as well as OTUs present in multiple oligotypes change over time. I will focus on these OTUs: 

OTU 4 (Cyanobium): It is an indicator of Oligotype 2, but also of September 9th.
OTU 312 (Reyranella): It is an indicator of Oligotype 2, but also of September 9th.
OTU 35 (Microscillaceae): It is an indicator of Oligotype 3, but also of July-August
OTU 32 (Microscillaceae): It is present on colonies from every date.
OTU 52 (Microscillaceae): It is frequently observed at high relative abundance
OTU 43 (Sutterellaceae): One of the most abundant and frequently observed OTUs
OTU 7 (Pseudanabaena): An abundant indicator of July-August
OTU 34 (Caulobacteraceae): An abundant indicator of July-August
OTU 10 (Roseomonas): An abundant indicator of July-August
OTU 11 (Pseudanabaena): An abundant indicator of July-August
OTU 46 (Tabrizicola): An abundant indicator of July-August

```{r}
#Add in the colony metadata to the OTU dataframe (so that we can separate the abundances by colony oligotype later on):
#Set all.y equal to false so that R doesn't try to add the metadata from the PBS blanks into the dataframe.
OTU_time_series_df <- merge(LE_H2O2_2019_all.merged.long, MC_2019.metadata,
                            by=c("Sample_ID", "Date_Collected", "Year", "Sample_Type",
                                 "Experiment", "Extraction_Date"), all.x=TRUE, all.y=FALSE)

#Average the OTU percent abundance across replicate filters:  
OTU_time_series_df <- OTU_time_series_df %>%
  group_by(Row.names, Date_Collected, Year, Station.x, Sample_Type, Oligotype) %>%
  summarise(n=n(), mean=mean(Perc_abund), sd=sd(Perc_abund)) %>%
  mutate(se=sd/sqrt(n)) %>%
  mutate(ci=se*1.96)

#Create a vector of OTUs of interest:
OTUs_of_interest <- c("Otu00004", "Otu00312", "Otu00035", "Otu00032", "Otu00052", "Otu00043", "Otu00007", "Otu00034", "Otu00010", "Otu00011", "Otu00046", "Otu00055")

#Convert the Date_Collected column into a date format. This allows the dates to be arranged in the correct temporal order.
OTU_time_series_df$Date_Collected <- dmy(OTU_time_series_df$Date_Collected)
#Remove year data from the Date_Collected column for so that the dates from multiple years are overlaid on the plot:
OTU_time_series_df$DayofYear <- yday(OTU_time_series_df$Date_Collected)

#Plot
OTU_time_series <- filter(OTU_time_series_df, Row.names %in% OTUs_of_interest & Sample_Type == "Whole Water") %>% 
  ggplot(aes(x=DayofYear, y=mean, color=Station.x)) +
    geom_line() +
    geom_point() +
    facet_grid(Row.names ~ as.factor(Year), scales="free_y") +
    scale_color_manual(values=c("coral", "dodgerblue", "chartreuse", "cadetblue4", "burlywood", "cyan3",
                                "gold", "darkorchid", "deeppink", "plum1", "slategray4", "navy", "red"),
                       name = "Station") +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.2, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.title = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0))) +
  xlab("") +
  ylab(expression("Percent Abundance"))

OTU_time_series
ggsave("OTU_time_series.pdf", plot = OTU_time_series, width = 8.5, height = 11, units = "in", dpi=320)
```

Let's compare transitions in the OTU abundances with time in the 2019 samples with average abundances in the colonies with time.  
```{r}
#Remove the Cyanobium OTU, which will be plotted separately
OTUs_of_interest <- c("Otu00312", "Otu00035", "Otu00032", "Otu00052", "Otu00043", "Otu00007", "Otu00034", "Otu00010", "Otu00011", "Otu00046", "Otu00055")
#Plot for whole water data
filt_OTU_time_series_2019 <- filter(OTU_time_series_df, Year == "2019" & Sample_Type == "filtered" & Row.names %in% OTUs_of_interest) %>%
  ggplot(aes(x=Date_Collected, y=mean)) +
    geom_line() +
    geom_point(aes(shape=Station.x)) +
    #geom_text(aes(label=Station.x), nudge_y=0.25, check_overlap = T) +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci)) +
    facet_grid(~Row.names, scales="free_y") +
    scale_color_manual(values=c("coral", "dodgerblue", "chartreuse", "cadetblue4", "burlywood", "cyan3",
                                "gold", "darkorchid", "deeppink", "plum1", "slategray4", "navy", "red"),
                       name = "Station") +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_blank(),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.1, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "top",
      legend.title = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0))) +
  scale_x_date(date_breaks = "2 weeks", date_labels = "%d %b %Y") +
  scale_y_continuous(breaks = seq(0,15, by=3)) +
  coord_cartesian(ylim=c(0,15)) +
  xlab("") +
  ylab(expression("Percent Abundance"))

filt_cyanobium_time_series_2019 <- filter(OTU_time_series_df, Year == "2019" & Sample_Type == "filtered" & Row.names == "Otu00004") %>%
  ggplot(aes(x=Date_Collected, y=mean)) +
    geom_line() +
    geom_point(aes(shape=Station.x)) +
    #geom_text(aes(label=Station.x), nudge_y=0.25, check_overlap = T) +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci)) +
    facet_grid(~Row.names, scales="free_y") +
    scale_color_manual(values=c("coral", "dodgerblue", "chartreuse", "cadetblue4", "burlywood", "cyan3",
                                "gold", "darkorchid", "deeppink", "plum1", "slategray4", "navy", "red"),
                       name = "Station") +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.1, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "none") +
  scale_x_date(date_breaks = "2 weeks", date_labels = "%d %b %Y") +
  xlab("") +
  ylab(expression("Percent Abundance"))

#Plot for colony data
Colony_OTU_time_series_2019 <- filter(OTU_time_series_df, Row.names %in% OTUs_of_interest & Sample_Type == "Colony") %>% 
  ggplot(aes(x=Date_Collected, y=mean, color=Oligotype)) +
    geom_line() +
    geom_point() +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci)) +
    facet_grid(~Row.names, scales="free_y") +
    scale_color_manual(values=c("coral", "dodgerblue", "plum1", "red"),
                       name = "Oligotype") +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_blank(),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.1, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "top",
      legend.title = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0))) +
  scale_x_date(date_breaks = "3 weeks", date_labels = "%d %b %Y") +
  scale_y_continuous(breaks = seq(0,60, by=20)) +
  coord_cartesian(ylim=c(0,60)) +
  xlab("Percent Abundance") +
  ylab(expression("Percent Abundance"))

Cyanobium_OTU_time_series_2019 <- filter(OTU_time_series_df, Row.names == "Otu00004" & Sample_Type == "Colony") %>% 
  ggplot(aes(x=Date_Collected, y=mean, color=Oligotype)) +
    geom_line() +
    geom_point() +
    geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci)) +
    facet_grid(~Row.names, scales="free_y") +
    scale_color_manual(values=c("coral", "dodgerblue", "plum1", "red"),
                       name = "Oligotype") +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.2, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "none") +
  scale_x_date(date_breaks = "3 weeks", date_labels = "%d %b %Y") +
  scale_y_continuous(breaks = seq(0,90, by=30)) +
  coord_cartesian(ylim=c(0,90)) +
  xlab("Percent Abundance") +
  ylab(expression("Percent Abundance"))

combo_OTU_time_series_2019 <- (filt_cyanobium_time_series_2019 + filt_OTU_time_series_2019 + plot_layout(widths = c(1,13))) / (Cyanobium_OTU_time_series_2019 + Colony_OTU_time_series_2019 + plot_layout(widths = c(1,13)))
ggsave("combo_OTU_time_series_2019.pdf", plot = combo_OTU_time_series_2019, width = 169, height = 150, units = "mm", dpi=600)
```

What is the percent composition of the microbes in the mock community?
```{r}
#Calculate the observation frequency of each OTU, add to dataframe:
otu_nonzero_counts <- apply(MC_2019.merged.mock[2], 1, function(y) sum(length(which(y > 0))))
MC_2019.merged.mock$otu_obs_freq <- otu_nonzero_counts

#Remove any OTUs not present in the colony samples:
MC_2019.merged.mock <- MC_2019.merged.mock[ MC_2019.merged.mock$otu_obs_freq !=0, ]

#Remove the OTUs with one read:
MC_2019.merged.mock <- MC_2019.merged.mock[ MC_2019.merged.mock$MC2019_Mock != 1, ]

#Remove the OTUs flagged as rare:
MC_2019.merged.mock <- MC_2019.merged.mock[ !(MC_2019.merged.mock$Row.names %in% rares_to_remove), ]

#Remove the OTUs flagged as contaminants:
MC_2019.merged.mock <- MC_2019.merged.mock[ !(MC_2019.merged.mock$Row.names %in% contaminants), ]

#Convert to percent abundance:
MC_2019.merged.mock[, 2] <- apply(MC_2019.merged.mock[,2,drop=F], 2, function(x) (x / sum(x)) *100 )
```

Microcystis = 49.4 %  
Bacillus = 29.7 %  
Stenotrophomonas = 20.8 %  

Added (copy number data is from rrnDB):  
1.75e7 cells Microcystis * 2 16S copies/cell = 3.5e7 copies/ml  
1.75e5 cells Bacillus * 5-17 16S copies/cell = 8.75e5 - 2.975e6 copies/mL  
1.75e6 cells/ml Stenotrophomonas * 2-4 copies/cell = 3.5e6 - 7e6 copies/ml  

Range of percentages:
Microcystis: 88.9% - 77.8%
Bacillus: 6.6-2.2%
Stenotrophomonas: 15.6 - 8.9%

We have way more Bacillus than we ought to, but we know there are Bacillus reads that are contaminants from the extraction kit, so this could explain some of the difference. Otherwise, we are 5% short of what we expect from Stenotrophomonas and 20% short for Microcystis. We over estimate Bacillus by about 20%.

Make a plot of chlorophyll a, phycocyanin, and particulate microcystin over time for WE8 and WE16:  
```{r}
#Import NOAA NCEI datatable
NCEI_df <- read.table("lake_erie_habs_field_sampling_results_2019.csv", header= TRUE, sep=",")

#Filter out entries from stations WE8 and WE16:
NCEI_df <- filter(NCEI_df, Site == "WE16" | Site == "WE8")

#Filter out entries from surface samples:
NCEI_df <- filter(NCEI_df, Depth_category == "Surface")

#Convert date column to date format:
NCEI_df$Date <- mdy(NCEI_df$Date)

#Make a line plots for each station:
PC_plot <- ggplot(NCEI_df, aes(x=Date, y=PC, color=Site)) +
  geom_line() +
  geom_point() +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.2, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "top",
      legend.title = element_blank()) +
  scale_x_date(date_breaks = "1 week", date_labels = "%d %b %Y") +
  coord_cartesian(ylim=c(0,100)) +
  xlab("") +
  ylab(expression("Phycocyanin ("*mu*"g/L)"))

Chla_plot <- ggplot(NCEI_df, aes(x=Date, y=Chla, color=Site)) +
  geom_line() +
  geom_point() +
    theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.2, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "none") +
  scale_x_date(date_breaks = "1 week", date_labels = "%d %b %Y") +
  coord_cartesian(ylim=c(0,100)) +
  xlab("") +
  ylab(expression("Chlorophyll a ("*mu*"g/L)"))

MC_plot <- ggplot(NCEI_df, aes(x=Date, y=Part_MC, color=Site)) +
  geom_line() +
  geom_point() +
          theme_classic() +
    theme(
      strip.text = element_text(size = 8),
      strip.background = element_rect(size = 0.25),
      panel.border=element_rect(colour="black",size=0.1,fill="NA"),
      axis.line = element_line(size=0.1),
      axis.title.x = element_blank(), 
      axis.title.y = element_text(size=8, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(size = 7, color = "black", angle = 45, hjust=1, margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.text.y = element_text(size = 7, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
      axis.ticks.length= unit(-0.2, "cm"),
      axis.ticks = element_line(size=0.1),
      legend.position = "none") +
    scale_x_date(date_breaks = "1 week", date_labels = "%d %b %Y") +
    xlab("") +
    ylab(expression("Particulate Microcystins ("*mu*"g/L)"))

Field_bloom_parameter_plot <- PC_plot / Chla_plot / MC_plot
Field_bloom_parameter_plot
ggsave("Field_bloom_parameter_plot.pdf", plot = Field_bloom_parameter_plot, width = 18, height = 20, units = "cm", dpi=320)
```

Amendment to ordination and ANOSIM analyses
---
In the above code, OTU percent abundance was calculated including the Microcystis sequences so that the contribution of the other bacteria to the total colony community could be assessed. Then Microcystis was removed prior to ordination analysis in an attempt to remove any impacts that variations in Microcystis abundance have on community composition. However, because the relative abundance was already calculated including Microcystis sequences, Microcystis abundance may still have been influencing the ordination of colonies by community dissimilarity. The below code corrects for this shortcoming by performing the ordination and ANOSIM tests with the OTU relative abundances calculated after Microcystis was removed from the datatables (not before).

Rerun the clustering analysis with only single colony samples, but remove Microcystis before converting to percent abundance:  
Data table generation and data wrangling is below:  
```{r}
contaminants <- c("Otu00038", "Otu04285", "Otu00209", "Otu00343", "Otu00375", "Otu00595", "Otu00602", "Otu00588", "Otu01608", "Otu02806" , "Otu04715", "Otu01416")

#Reprocess the data frame, removing the contaminant OTUs:
#Load dataframes:
MC_2019.shared <- read.table("MC_2019.shared", header = TRUE, sep="\t")
MC_2019.taxonomy <- read.table("LE_H2O2.taxonomy", header = TRUE, sep="\t")
MC_2019.metadata <- read.table("MC_2019_metadata.txt", header= TRUE, sep="\t")

#Fix up OTU table and merge with taxonomy table:
rownames(MC_2019.shared) <- MC_2019.shared$Group
drop <- c("label", "numOtus", "Group")
MC_2019.shared <- MC_2019.shared[ , !(names(MC_2019.shared) %in% drop)]
MC_2019.shared <- t(MC_2019.shared) #transpose table so that OTUs are row names for merging
MC_2019.shared <- MC_2019.shared[rowSums(MC_2019.shared) != 0,] #Remove Otus that are absent from colony samples
rownames(MC_2019.taxonomy) <- MC_2019.taxonomy$OTU
drop <- "OTU"
MC_2019.taxonomy <- MC_2019.taxonomy[ , !(names(MC_2019.taxonomy) %in% drop)]
MC_2019.taxonomy <- MC_2019.taxonomy[ MC_2019.taxonomy$Size > 2, ] #Only keep OTUs that have more than 2 sequences
MC_2019.merged <- merge(MC_2019.shared, MC_2019.taxonomy, by="row.names", all = FALSE)
#Divide the taxonomy into separate columns for each rank for sorting
MC_2019.merged <- separate(MC_2019.merged, Taxonomy, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), sep=";", remove=TRUE)

#Filter dataframe so that we save the control samples into a separate dataframe for later
keep <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Sample_Type == "PBS Blank" ] 
taxcols <- c("Row.names", "Size", "Domain", "Phylum", "Class", "Order", "Family", "Genus")
MC_2019.merged.controls <- MC_2019.merged[ , names(MC_2019.merged) %in% keep | names(MC_2019.merged) %in% taxcols]

#Remove samples that were controls and had a low number of reads:
drop <- MC_2019.metadata$Sample_ID[ MC_2019.metadata$Enough_reads == "No" ]
MC_2019.merged <- MC_2019.merged[ , !(names(MC_2019.merged) %in% drop)]
MC_2019.merged.mock <- MC_2019.merged[ , names(MC_2019.merged) == "MC2019_Mock" | names(MC_2019.merged) == "Row.names"]

#Remove the Mock Sample:
MC_2019.merged <- MC_2019.merged[ , names(MC_2019.merged) != "MC2019_Mock"]

#Calculate the observation frequency of each OTU, add to dataframe:
otu_nonzero_counts <- apply(MC_2019.merged[2:45], 1, function(y) sum(length(which(y > 0))))
otu_obs_freq <- ( otu_nonzero_counts / 44 ) * 100
MC_2019.merged$otu_obs_freq <- otu_obs_freq

#Remove any OTUs not present in the colony samples:
MC_2019.merged <- MC_2019.merged[ MC_2019.merged$otu_obs_freq !=0, ]
Num_otus_prefilter <- nrow(MC_2019.merged)

#Remove the OTUs flagged as rare:
MC_2019.merged <- MC_2019.merged[ !(MC_2019.merged$Row.names %in% rares_to_remove), ]

#Remove the OTUs flagged as contaminants:
MC_2019.merged <- MC_2019.merged[ !(MC_2019.merged$Row.names %in% contaminants), ]
Num_otus_postfilter <- nrow(MC_2019.merged)

#How many OTUs were removed?
Num_otus_prefilter - Num_otus_postfilter

#What is the percentage of OTUs kept?
(Num_otus_postfilter / Num_otus_prefilter) * 100

#Save a copy of the df with raw counts for rarefaction curves later:
Rarefaction_df <- MC_2019.merged

#Save a copy of the data frame without the Microcystis OTUs before normalizing:
MC_2019.merged.noMa <- MC_2019.merged[MC_2019.merged$Genus != "Microcystis_PCC-7914", ]

#Convert to percent abundance:
MC_2019.merged.noMa[, 2:45] <- apply(MC_2019.merged.noMa[,2:45], 2, function(x) (x / sum(x)) *100 )

#Add the maximum, mean, and median relative abundance for each OTU as a column in the dataframe (not including zeros):
MC_2019.merged.noMa.nonzeros <- MC_2019.merged.noMa
MC_2019.merged.noMa.nonzeros[MC_2019.merged.noMa.nonzeros == 0] <- NA
MC_2019.merged.noMa$Max_perc_abund <- apply(MC_2019.merged.noMa[2:45], 1, max)
MC_2019.merged.noMa$Median_perc_abund <- apply(MC_2019.merged.noMa.nonzeros[2:45], 1, median, na.rm=TRUE)
MC_2019.merged.noMa$Mean_perc_abund <- apply(MC_2019.merged.noMa.nonzeros[2:45], 1, mean, na.rm=TRUE)

#Convert dataframe to long format:
MC_2019.merged.noMa.long <- gather(MC_2019.merged.noMa, Sample_ID, Perc_abund, 2:45)

#Add colony metadata:
MC_2019.merged.noMa.long <- merge(MC_2019.merged.noMa.long, MC_2019.metadata, by="Sample_ID", all.x = TRUE)
drop <- c("Sample_Type", "Amplification", "Enough_reads") #remove these columns, because they are no longer useful
MC_2019.merged.noMa.long <- MC_2019.merged.noMa.long[, !(names(MC_2019.merged.noMa.long) %in% drop) ]

#Fix up phylum names for aesthetic purposes:
MC_2019.merged.noMa.long$Phylum <- gsub("Acidobacteriota", "Acidobacteria", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Actinobacteriota", "Actinobacteria", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Bacteria_unclassified", "Unclassified Bacteria", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Bacteroidota", "Bacteroidetes", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Cloacimonadota", "Cloacimonetes", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Deinococcota", "Deinococcus", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Gemmatimonadota", "Gemmatimonadetes", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Nitrospirota", "Nitrospirae", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Planctomycetota", "Planctomycetes", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Spirochaetota", "Spirochaetes", MC_2019.merged.noMa.long$Phylum)
MC_2019.merged.noMa.long$Phylum <- gsub("Verrucomicrobiota", "Verrucomicrobia", MC_2019.merged.noMa.long$Phylum)
```

```{r}
#Take our processed OTU table and remove the metadata columns:
#Columns 1-47 contain the info we need, the rest is metadata
MC_2019.matrix.noMa <- MC_2019.merged.noMa[MC_2019.merged.noMa$Genus != "Microcystis_PCC-7914",1:45] #Remove the Microcystis Otu
rownames(MC_2019.matrix.noMa) <- MC_2019.matrix.noMa$Row.names
MC_2019.matrix.noMa <- MC_2019.matrix.noMa[ , colnames(MC_2019.matrix.noMa) != "Row.names" ]
#Transpose the matrix so that samples are rows and OTUs are columns:
MC_2019.matrix.noMa <- t(MC_2019.matrix.noMa)

#Build the plot using the metaNMDS function in vegan with three axes:
set.seed(123)
NMDS_3.noMa <- metaMDS(MC_2019.matrix.noMa, distance = "bray", k=3, autotransform = FALSE, try = 40, trymax = 100)
```
The Stress value has decreased from 0.12 to 0.11.  

Next make a stress plot:  
```{r}
stressplot(NMDS_3.noMa, xlim=c(0,1))
```

Dissimilarity increases with ordination distance as expected.  Plot the ordination:  
```{r}
#Extract the axis coordinates (the NMDS scores):
NMDS_3.noMa.scores <- as.data.frame(scores(NMDS_3.noMa))

#Add metadata:
NMDS_3.noMa.scores <- merge(NMDS_3.noMa.scores, MC_2019.metadata, by.x = "row.names", by.y = "Sample_ID", all.x = TRUE)
rownames(NMDS_3.noMa.scores) <- NMDS_3.noMa.scores$Row.names
drop <- "Row.names"
NMDS_3.noMa.scores <- NMDS_3.noMa.scores[ , !(names(NMDS_3.noMa.scores) %in% drop) ]

#Plot axis 1 vs axis 2:
NMDS_3.1v2.noMa.plot <- ggplot(NMDS_3.noMa.scores, aes(x=NMDS1, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Oligotype, colour=Date_Collected)) +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 2") +
  xlab("NMDS 1")

#Plot axis 1 vs axis 3:
NMDS_3.1v3.noMa.plot <- ggplot(NMDS_3.noMa.scores, aes(x=NMDS3, y=NMDS1)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Oligotype, colour=Date_Collected)) +
  theme_classic() +
  theme(plot.title = element_text(hjust=1, size=10),
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "none") +
  ylab("NMDS 1") +
  xlab("NMDS 3")

#Plot axis 2 vs axis 3:
NMDS_3.2v3.noMa.plot <- ggplot(NMDS_3.noMa.scores, aes(x=NMDS3, y=NMDS2)) +
  geom_point(size = 3, alpha=0.8, aes(shape=Oligotype, colour=Date_Collected)) +
  ggtitle("Stress = 0.11") +
  theme_classic() +
  theme(
    axis.line.x = element_line(size=0.1),
    axis.line.y = element_line(size=0.1),
    axis.title.x = element_text(size=12, margin = margin(t = 10, r = 0, b = 0, l = 0)), 
    axis.title.y = element_text(size=12, margin = margin(t = 0, r = 10, b = 0, l = 0)),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(size = 10, color = "black", margin = margin(t = 10, r = 0, b = 0, l = 0)),
    axis.text.y = element_text(size = 10, color = "black", margin = margin(t = 0, r = 10, b = 0, l = 0)),
    axis.ticks.length = unit(-0.1, "cm"),
    axis.ticks = element_line(size = 0.1),
    legend.position = "bottom",
    legend.box = "vertical",
    legend.text = element_text(size = 10), legend.title = element_text(size = 12)) +
  ylab("NMDS 2") +
  xlab("NMDS 3")

NDMS3.noMa <- NMDS_3.1v2.noMa.plot + NMDS_3.2v3.noMa.plot + guide_area() + NMDS_3.1v3.noMa.plot + plot_layout(nrow = 2, guide="collect")
NDMS3.noMa <- NDMS3.noMa + labs(colour = "Sampling Date")
NDMS3.noMa
ggsave("NDMS3.noMa.pdf", NDMS3.noMa, width = 169, height = 180, units = "mm", dpi=600)
```

Let's run the ANOSIM tests to check for significant differences in the mean dissimilarities between sampling dates and morphotype:  
```{r}
#ANOSIM for collection date:
ano_date <- anosim(MC_2019.matrix.noMa, NMDS_3.noMa.scores$Date_Collected, distance = "bray", permutations = 9999)
#ANOSIM for colony morphology:
ano_morph <- anosim(MC_2019.matrix.noMa, NMDS_3.noMa.scores$Morphotype, distance = "bray", permutations = 9999)
#ANOSIM for oligotype:
ano_oligo <- anosim(MC_2019.matrix.noMa, NMDS_3.noMa.scores$Oligotype, distance = "bray", permutations = 9999)

ano_date
ano_morph
ano_oligo
```
Same conclusion as before, but the separation by oligotype is now stronger than previously calculated. ANOSIM R (oligotype) increased from 0.49 to 0.53 (not much change). The ANOSIM R (Sampling Date) increased from 0.41 to 0.43 (not much change). The correlation is significant with both normalization methods. 

How does the hierarchical clustering differ:  
```{r}
#Make a distance matrix:
MC_2019_all.dist.noMa <- vegdist(MC_2019.matrix.noMa, method="bray")
#Cluster using hierarchical clustering
MC_2019_all.clust.noMa <- hclust(MC_2019_all.dist.noMa, method = "average")
#Get groups from the hclust dendogram
MC_2019_all.groups.noMa <- cutree(MC_2019_all.clust.noMa, k=4)

#Convert clustering results to a dendogram format:
MC_2019_all.dendro.noMa <- as.dendrogram(MC_2019_all.clust.noMa)
MC_2019_all.dendro.noMa <- color_branches(MC_2019_all.dendro.noMa, k=4, col = c("black"), groupLabels = TRUE)
MC_2019_all.dendro.noMa <- set(MC_2019_all.dendro.noMa, "labels_cex", 0.5)
#Plot
Colony_dendrog_plot.noMa <- plot(MC_2019_all.dendro.noMa, ylab = "Bray-Curtis Dissimilarity")
```
Only two colonies were put in different HC clusteres, the new assignments better fit general patterns in oligotype and sampling data assignment to HC clusters.  